<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">

<!--Converted with LaTeX2HTML 99.2beta8 (1.46)
original version by:  Nikos Drakos, CBLU, University of Leeds
* revised and updated by:  Marcus Hennecke, Ross Moore, Herb Swan
* with significant contributions from:
  Jens Lippmann, Marek Rouchal, Martin Wilck and others -->
<HTML>
<HEAD>
<TITLE>Contents</TITLE>
<META NAME="description" CONTENT="Contents">
<META NAME="keywords" CONTENT="sbml-level-1">
<META NAME="resource-type" CONTENT="document">
<META NAME="distribution" CONTENT="global">

<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=iso-8859-1">
<META NAME="Generator" CONTENT="LaTeX2HTML v99.2beta8">
<META HTTP-EQUIV="Content-Style-Type" CONTENT="text/css">

<LINK REL="STYLESHEET" HREF="sbml-level-1.css">

</HEAD>

<BODY  BGCOLOR="#ffffff">

<P>

<P>

<P>

<P>

<P>

<P>

<P>

<P>
<HR NOSHADE>
<H1 ALIGN="CENTER">Systems Biology Markup Language (SBML) Level 1:
<BR>  Structures and Facilities for Basic Model Definitions</H1>
<HR NOSHADE>
<P ALIGN="CENTER"><STRONG>Michael Hucka, Andrew Finney, Herbert Sauro, Hamid Bolouri</STRONG></P>
<P ALIGN="CENTER"><TT>{mhucka,afinney,hsauro,hbolouri}@cds.caltech.edu</TT></P>
<P ALIGN="CENTER"><SMALL>Systems Biology Workbench Development Group
<BR>  ERATO Kitano Systems Biology Project
<BR>  Control and Dynamical Systems, MC 107-81
<BR>  California Institute of Technology, Pasadena, CA 91125, USA<BR>  <TT><A NAME="tex2html1"
  HREF="http://www.cds.caltech.edu/erato">http://www.cds.caltech.edu/erato</A></TT></SMALL></P>
<P ALIGN="CENTER"><SMALL>Principal Investigators: John Doyle and Hiroaki Kitano</SMALL></P>
<P ALIGN="CENTER"><STRONG>SBML Level 1, Version 1 (Final), 2 March 2001</STRONG></P><BR>

<H2><A NAME="SECTION00000000000000000000">
Contents</A>
</H2>
<!--Table of Contents-->

<UL>
<LI><A NAME="tex2html83"
  HREF="sbml-level-1.html">1 Introduction</A>
<UL>
<LI><A NAME="tex2html84"
  HREF="#SECTION00011000000000000000">1.1 Scope and Limitations</A>
<LI><A NAME="tex2html85"
  HREF="#SECTION00012000000000000000">1.2 Notational Conventions</A>
</UL>
<LI><A NAME="tex2html86"
  HREF="#SECTION00020000000000000000">2 Overview of SBML</A>
<LI><A NAME="tex2html87"
  HREF="#SECTION00030000000000000000">3 Preliminary Definitions</A>
<UL>
<LI><A NAME="tex2html88"
  HREF="#SECTION00031000000000000000">3.1 Type <TT>SBase</TT></A>
<LI><A NAME="tex2html89"
  HREF="#SECTION00032000000000000000">3.2 Guidelines for the Use of the <TT>annotations</TT> Field in
  <TT>SBase</TT></A>
<LI><A NAME="tex2html90"
  HREF="#SECTION00033000000000000000">3.3 Type <TT>SName</TT></A>
<LI><A NAME="tex2html91"
  HREF="#SECTION00034000000000000000">3.4 Component Names and Namespaces in SBML</A>
<LI><A NAME="tex2html92"
  HREF="#SECTION00035000000000000000">3.5 Formulas</A>
</UL>
<LI><A NAME="tex2html93"
  HREF="#SECTION00040000000000000000">4 SBML Components</A>
<UL>
<LI><A NAME="tex2html94"
  HREF="#SECTION00041000000000000000">4.1 Models</A>
<LI><A NAME="tex2html95"
  HREF="#SECTION00042000000000000000">4.2 Unit Definitions</A>
<LI><A NAME="tex2html96"
  HREF="#SECTION00043000000000000000">4.3 Compartments</A>
<LI><A NAME="tex2html97"
  HREF="#SECTION00044000000000000000">4.4 Species</A>
<LI><A NAME="tex2html98"
  HREF="#SECTION00045000000000000000">4.5 Parameters</A>
<LI><A NAME="tex2html99"
  HREF="#SECTION00046000000000000000">4.6 Rules</A>
<LI><A NAME="tex2html100"
  HREF="#SECTION00047000000000000000">4.7 Reactions</A>
</UL>
<LI><A NAME="tex2html101"
  HREF="#SECTION00050000000000000000">5 Examples of Full Models Encoded in XML Using SBML</A>
<UL>
<LI><A NAME="tex2html102"
  HREF="#SECTION00051000000000000000">5.1 A Simple Example Application of SBML</A>
<LI><A NAME="tex2html103"
  HREF="#SECTION00052000000000000000">5.2 Simple Use of Units Feature in a Model</A>
<LI><A NAME="tex2html104"
  HREF="#SECTION00053000000000000000">5.3 A Simple Example Application Using Rules</A>
</UL>
<LI><A NAME="tex2html105"
  HREF="#SECTION00060000000000000000">6 Discussion</A>
<UL>
<LI><A NAME="tex2html106"
  HREF="#SECTION00061000000000000000">6.1 Future Enhancements to SBML: Level 2 and Beyond</A>
<LI><A NAME="tex2html107"
  HREF="#SECTION00062000000000000000">6.2 Relationships to Other Efforts</A>
<LI><A NAME="tex2html108"
  HREF="#SECTION00063000000000000000">6.3 Tracking the XML Schema Standard</A>
<LI><A NAME="tex2html109"
  HREF="#SECTION00064000000000000000">6.4 Availability</A>
</UL>
<LI><A NAME="tex2html110"
  HREF="#SECTION00070000000000000000">7 Acknowledgments</A>
<LI><A NAME="tex2html111"
  HREF="#SECTION00080000000000000000">8 Appendix</A>
<LI><A NAME="tex2html112"
  HREF="#SECTION00090000000000000000">A. Summary of Notation</A>
<LI><A NAME="tex2html113"
  HREF="#SECTION000100000000000000000">B. XML Schema for SBML</A>
<LI><A NAME="tex2html114"
  HREF="#SECTION000110000000000000000">C. Predefined Functions in SBML</A>
<LI><A NAME="tex2html115"
  HREF="#SECTION000120000000000000000">Bibliography</A>
<LI><A NAME="tex2html116"
  HREF="#SECTION000130000000000000000">About this document ...</A>
</UL>
<!--End of Table of Contents-->
<H1><A NAME="SECTION00010000000000000000"></A>
<A NAME="sec:introduction"></A><BR>
1 Introduction
</H1>

<P>
We present the <B>S</B>ystems <B>B</B>iology <B>M</B>arkup
<B>L</B>anguage (SBML) Level&nbsp;1, a description language for simulations in
systems biology.  SBML is oriented towards representing biochemical
networks common in research on a number of topics, including cell signaling
pathways, metabolic pathways, biochemical reactions, gene regulation, and
many others.  A recent conference&nbsp;(<A
 HREF="sbml-level-1.html#kitano:2001">Kitano, 2001</A>) highlights the range
of topics that fall under the umbrella of <I>systems biology</I> and are in
the domain of the description language defined here.  Many contemporary
research initiatives demonstrate the growing popularity of this kind of
multidisciplinary
work&nbsp;(e.g., <A NAME="tex2html117"
  HREF="#gilman:2000">Gilman, 2000</A>; <A NAME="tex2html118"
  HREF="#abbott:1999">Abbott, 1999</A>; <A NAME="tex2html119"
  HREF="#popel:1998">Popel and Winslow, 1998</A>; <A NAME="tex2html120"
  HREF="#smaglik:2000">Smaglik, 2000a</A>,<A NAME="tex2html121"
  HREF="#smaglik:2000b">b</A>).

<P>
SBML Level&nbsp;1 is the result of merging modeling-language features from the
following simulation systems: <I>BioSpice</I>&nbsp;(<A
 HREF="sbml-level-1.html#arkin:2001">Arkin, 2001</A>),
<I>DBSolve</I>&nbsp;(<A NAME="tex2html122"
  HREF="#goryanin:2001">Goryanin, 2001</A>; <A NAME="tex2html123"
  HREF="#goryanin:1999">Goryanin et&nbsp;al., 1999</A>),
<I>E-Cell</I>&nbsp;(<A NAME="tex2html124"
  HREF="#tomita:2001">Tomita et&nbsp;al., 2001</A>,<A NAME="tex2html125"
  HREF="#tomita:1999">1999</A>),
<I>Gepasi</I>&nbsp;(<A NAME="tex2html126"
  HREF="#mendes:2001">Mendes, 2001</A>,<A NAME="tex2html127"
  HREF="#mendes:1997">1997</A>),
<I>Jarnac</I>&nbsp;(<A NAME="tex2html128"
  HREF="#sauro:2000">Sauro, 2000</A>; <A NAME="tex2html129"
  HREF="#sauro:1991">Sauro and Fell, 1991</A>),
<I>StochSim</I>&nbsp;(<A NAME="tex2html130"
  HREF="#morton-firth:1998">Morton-Firth and Bray, 1998</A>; <A NAME="tex2html131"
  HREF="#bray:2001">Bray et&nbsp;al., 2001</A>), and <I>Virtual
  Cell</I>&nbsp;(<A NAME="tex2html132"
  HREF="#schaff:2000">Schaff et&nbsp;al., 2000</A>,<A NAME="tex2html133"
  HREF="#schaff:2001">2001</A>).  SBML was developed with the help
of the authors of these packages.  As a result of being based on actual
working simulation software, it is a practical and functional description
language.  Our goal in creating it has been to provide an open standard
that will enable simulation software to exchange models, something that is
currently impossible because there is no standard model exchange language.
We expect SBML models to be encoded using XML, the eXtensible Markup
Language&nbsp;(<A NAME="tex2html134"
  HREF="#bray:1998">Bray et&nbsp;al., 1998</A>; <A NAME="tex2html135"
  HREF="#bosak:1999">Bosak and Bray, 1999</A>), and we include here an XML Schema
that defines SBML Level&nbsp;1.

<P>

<H2><A NAME="SECTION00011000000000000000">
1.1 Scope and Limitations</A>
</H2>

<P>
SBML Level 1 is meant to support non-spatial biochemical models and the
kinds of operations that are possible in existing analysis/simulation
tools.  A number of potentially desirable features have been intentionally
omitted from the language definition.  Future software tools will
undoubtedly require the evolution of SBML; we expect that subsequent
releases of SBML (termed <I>levels</I>) will add additional structures and
facilities currently missing from Level&nbsp;1, once the simulation community
gains experience with the current language definition.  In
Section&nbsp;<A HREF="sbml-level-1.html#sec:level-2">6.1</A>, we discuss extensions that will likely be
included in SBML Level&nbsp;2.

<P>
The definition of the model description language presented here does not
specify <I>how</I> programs should communicate or read/write SBML.  We
assume that for a simulation program to communicate a model encoded in
SBML, the program will have to translate its internal data structures to
and from SBML, use a suitable transmission medium and protocol, etc., but
these issues are outside of the scope of this document.

<P>

<H2><A NAME="SECTION00012000000000000000">
1.2 Notational Conventions</A>
</H2>

<P>
SBML is intended to be a common XML-based format for encoding systems
biology models in a simple form that software tools can use as an exchange
format.  However, for easier communication to human readers, we define SBML
using a graphical notation based upon UML, the Unified Modeling
Language&nbsp;(<A NAME="tex2html136"
  HREF="#oestereich:1999">Oestereich, 1999</A>; <A NAME="tex2html137"
  HREF="#eriksson:1998">Eriksson and Penker, 1998</A>).  This UML-based definition
in turn is used to define an XML
Schema&nbsp;(<A NAME="tex2html138"
  HREF="#biron:2000">Biron and Malhotra, 2000</A>; <A NAME="tex2html139"
  HREF="#thompson:2000">Thompson et&nbsp;al., 2000</A>; <A NAME="tex2html140"
  HREF="#fallside:2000">Fallside, 2000</A>) for SBML.  There are
three main advantages to using UML as a basis for defining SBML data
structures.  First, compared to using other notations or a programming
language, the UML visual representations are generally easier to grasp by
readers who are not computer scientists.  Second, the visual notation is
implementation-neutral: the defined structures can be encoded in any
concrete implementation language--not just XML, but C or Java as well.
Third, UML is a de facto industry standard that is documented in many
sources.  Readers are therefore more likely to be familiar with it than
other notations.

<P>
Our notation and our approach for mapping UML to XML Schemas is explained
in a separate document&nbsp;(<A
 HREF="sbml-level-1.html#hucka:2000b">Hucka, 2000</A>).  A summary of the essential
points is presented in Appendix&nbsp;<A HREF="sbml-level-1.html#apdx:notation">A</A>, and examples
throughout this document illustrate the approach.  We also follow certain
naming and typographical conventions throughout this document.
Specifically, the names of data structure attributes or fields begin with a
lowercase letter, and the names of data structures and types begin with an
uppercase letter.  Keywords (names of types, XML elements, etc.) are
written in a typewriter-style font; for example, <TT>Compartment</TT> is a
type name and <TT>compartment</TT> is a field name.  Likewise, literal XML
examples are also written in a typewriter-style font.

<P>

<H1><A NAME="SECTION00020000000000000000"></A>
<A NAME="sec:overview"></A><BR>
2 Overview of SBML
</H1>

<P>
The following is an example of a simple, hypothetical biochemical network that
can be represented in SBML:
<P></P>
<DIV ALIGN="CENTER"><!-- MATH
 \begin{equation*}
\begin{array}{@{}ccc@{}}
    X_0 & \underrightarrow{k_1 X_0} & S_1 \\\\[-4pt]
    S_1 & \underrightarrow{k_2 S_1} & X_1 \\\\[-4pt]
    S_1 & \underrightarrow{k_3 S_1} & X_2
  \end{array}
\end{equation*}
 -->
<TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="107" HEIGHT="113" BORDER="0"
 SRC="img11.gif"
 ALT="\begin{displaymath}\begin{array}{@{}ccc@{}} X_0 &amp; \underrightarrow{k_1 X_0} &amp; S_...
...\  \\  [-4pt] S_1 &amp; \underrightarrow{k_3 S_1} &amp; X_2 \end{array}\end{displaymath}"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>

<P>
Broken down into its constituents, this model contains a number of
components: reactant species, product species, reactions, rate laws, and
parameters in the rate laws.  To analyze or simulate this network,
additional components must be made explicit, including compartments for the
species, and units on the various quantities.  The top level of an SBML
model definition simply consists of lists of these components:
<DIV ALIGN="CENTER"><TABLE CELLPADDING=3>
<TR><TD ALIGN="CENTER"><TABLE  WIDTH="346">
<TR><TD>
      <PRE><TT>
        beginning of model definition
<BR>list of unit definitions (optional)
<BR>list of compartments
<BR>list of species
<BR>list of parameters (optional)
<BR>list of rules (optional)
<BR>list of reactions
<BR>end of model definition      
</TT></PRE></TD></TR>
</TABLE></TD>
</TR>
</TABLE></DIV>
The meaning of each component is as follows:
<DL>
<DT></DT>
<DD><I>Unit definition</I>: A name for a unit used in the expression of
  quantities in a model.  Units may be supplied in a number of contexts in
  an SBML model, and it is convenient to have a facility for both setting
  default units and for allowing combinations of units to be given
  abbreviated names.

<P>
</DD>
<DT></DT>
<DD><I>Compartment</I>: A container of finite volume for substances.  In
  SBML Level&nbsp;1, a compartment is primarily a topological structure with a
  volume but no geometric qualities.

<P>
</DD>
<DT></DT>
<DD><I>Specie</I>: A substance or entity that takes part in a reaction.
  Some example species are ions such as <!-- MATH
 $\text{Ca}^{2++}$
 -->
Ca<IMG
 WIDTH="31" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img12.gif"
 ALT="$ ^{2++}$"> and molecules
  such as glucose or ATP.  The primary qualities associated with a specie
  in SBML Level&nbsp;1 are its initial amount and the compartment in which it is
  located.

<P>
</DD>
<DT></DT>
<DD><I>Reaction</I>: A statement describing some transformation,
  transport or binding process that can change the amount of one or more
  species.  For example, a reaction may describe how certain entities
  (reactants) are transformed into certain other entities (products).
  Reactions have associated rate laws describing how quickly they take
  place.

<P>
</DD>
<DT></DT>
<DD><I>Parameter</I>: A quantity that has a symbolic name.  SBML Level&nbsp;1
  provides the ability to define parameters that are global to a model as
  well as parameters that are local to a single reaction.

<P>
</DD>
<DT></DT>
<DD><I>Rule</I>: In SBML, a mathematical expression that is added
  to the differential equations constructed from the set of reactions and
  can be used to set parameter values, establish constraints between
  quantities, etc.

<P>
</DD>
</DL>

<P>
A software package can read an SBML model description and translate it
into its own internal format for model analysis.  For example, a package
might provide the ability to simulate the model, by constructing
differential equations representing the network and then performing
numerical time integration on the equations to explore the model's dynamic
behavior.  

<P>
SBML allows models of arbitrary complexity to be represented.  Each type of
component in a model is described using a specific type of data structure
that organizes the relevant information.  The data structures determine how
the resulting model is encoded in XML.

<P>
In the sections that follow, the various constructs in SBML and their uses
are described in detail.  Section&nbsp;<A HREF="sbml-level-1.html#sec:general">3</A> first introduces a few
basic structures that are used throughout SBML, then
Section&nbsp;<A HREF="sbml-level-1.html#sec:elements">4</A> provides details on each of the main components
of SBML.  Section&nbsp;<A HREF="sbml-level-1.html#sec:xml-rep">5</A> provides several complete examples of
models encoded in XML using SBML.

<P>

<H1><A NAME="SECTION00030000000000000000"></A>
<A NAME="sec:general"></A><BR>
3 Preliminary Definitions
</H1>

<P>
This section covers certain constructs that are used repeatedly in the rest
of SBML and are useful to discuss before diving into the details of the
components provided in SBML.

<P>

<H2><A NAME="SECTION00031000000000000000"></A>
<A NAME="sec:sbase"></A><BR>
3.1 Type <TT>SBase</TT>
</H2>

<P>
Each of the main components composing an SBML model definition has a
specific data type that is derived directly or indirectly from a single
base type called <TT>SBase</TT>.  This inheritance hierarchy is depicted in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:top-level">1</A>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:top-level"></A><A NAME="1199"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 1:</STRONG>
A UML diagram of the inheritance hierarchy of major data types
    in SBML.  Open arrows indicate inheritance, pointing from inheritors to 
    their parents&nbsp;(<A NAME="tex2html141"
  HREF="#oestereich:1999">Oestereich, 1999</A>; <A NAME="tex2html142"
  HREF="#eriksson:1998">Eriksson and Penker, 1998</A>).</CAPTION>
<TR><TD><IMG
 WIDTH="605" HEIGHT="282" BORDER="0"
 SRC="img13.gif"
 ALT="\begin{figure}\vspace*{8pt}
\centering \incfig[scale = 0.7]{top-level}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
The type <TT>SBase</TT> is designed to allow a modeler or a software package
to attach information to each component in an SBML model.  The definition
of <TT>SBase</TT> is presented in Figure&nbsp;<A HREF="sbml-level-1.html#fig:identified">2</A>.
<TT>SBase</TT> contains two fields, both of which are optional:
<TT>notes</TT> and <TT>annotations</TT>.  The field <TT>notes</TT> is a
container for XHTML content.  It is intended for recording optional
user-visible annotations.  Every data object derived directly or indirectly
from type <TT>SBase</TT> can have a separate value for <TT>notes</TT>,
allowing users considerable freedom for annotating their models.  The
second field, <TT>annotations</TT>, is provided for software-generated
annotations.  It is a container for arbitrary data (XML type <TT>any</TT>)
and is intended to store information not intended for human viewing.  As
with the user-visible <TT>notes</TT> field, every data object can have its
own <TT>annotations</TT> value.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:identified"></A><A NAME="1200"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 2:</STRONG>
The definition of <TT>SBase</TT>.  Text enclosed in braces next
    to attribute types (i.e., <TT>{minOccurs="1"}</TT>) indicates
    constraints on the possible attribute values; we use XML Schema
    language to express constraints since we are primarily interested in
    the XML encoding of SBML.</CAPTION>
<TR><TD><IMG
 WIDTH="224" HEIGHT="58" BORDER="0"
 SRC="img14.gif"
 ALT="\begin{figure}\centering \incfig[scale = 0.68]{identified}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
In other type definitions presented below, we follow the UML convention of
eliding the attributes derived from a parent type such as <TT>SBase</TT>.
It should be kept in mind that these attributes are always available.

<P>

<H2><A NAME="SECTION00032000000000000000">
3.2 Guidelines for the Use of the <TT>annotations</TT> Field in
  <TT>SBase</TT></A>
</H2> 

<P>
The <TT>annotations</TT> field in the definition of <TT>SBase</TT> is
formally unconstrained in order that software developers may attach any
information they need to different components in an SBML model.  However,
it is important that this facility not be misused accidentally.  In
particular, it is critical that information essential to a model definition
is <I>not</I> stored in <TT>annotations</TT>.  Parameter values, functional
dependencies between model components, etc., should not be recorded as
annotations.

<P>
Here are examples of the kinds of data that may be appropriately stored in
<TT>annotations</TT>: (a) Information about graphical layout of model
components; (b) application-specific processing instructions that do not
change the essence of a model; (c) bibliographic information pertaining to
a given model; and (d) identification information for cross-referencing
components in a model with items in a database.  (We expect to introduce an
explicit scheme for recording bibliographic information and making database
references in SBML Level&nbsp;2, at which time using annotations for these
purposes will become unnecessary.)

<P>
Different applications may use XML Namespaces&nbsp;(<A
 HREF="sbml-level-1.html#bray:1999">Bray et&nbsp;al., 1999</A>) to specify
the intended vocabulary of a particular annotation.  Here is an example of
this kind of usage.  Suppose that a particular application wants to
annotate data structures in an SBML model definition with screen layout
information and a time stamp.  The application developers should choose a
URI (<I>Universal Resource Identifier</I>; <A NAME="tex2html143"
  HREF="#w3c:2000">W3C 2000a</A>; <A NAME="tex2html144"
  HREF="#harold:2001">Harold and Means 2001</A>)
reference that uniquely identifies the vocabulary that the
application will use for such annotations, and a prefix string to be used
in the annotations.  For illustration purposes, let us say the URI
reference is ``<TT>http://www.mysim.org/ns</TT>'' and the prefix is
<TT>mysim</TT>.  An example of an annotation might then be as follows:

<P>
<BR>
<IMG
 WIDTH="502" HEIGHT="76" ALIGN="BOTTOM" BORDER="0"
 SRC="img15.gif"
 ALT="\begin{example}
...
&lt;annotations xmlns:mysim=''http://www.mysim.org/ns''&gt;
&lt;mysi...
...imestamp&gt;2000-12-18 18:31 PST&lt;/mysim:timestamp&gt;
&lt;/annotations&gt;
...
\end{example}">
<BR>

<P>
The namespace prefix <TT>mysim</TT> is used to qualify the XML elements
<TT>mysim:nodecolors</TT> and <TT>mysim:timestamp</TT>; presumably these
symbols have meaning to the application.  This example places the XML
Namespace information on <TT>annotations</TT> itself rather than on a
higher-level enclosing construct or the enclosing document level, but other
placements would be valid as well&nbsp;(<A
 HREF="sbml-level-1.html#bray:1999">Bray et&nbsp;al., 1999</A>).

<P>
The use of XML Namespaces permits multiple applications to place
annotations on XML elements of a model without risking interference or
element name collisions.  Annotations stored by different simulation
packages can thus coexist in the same model definition.  Although XML
Namespace names (``<TT>http://www.mysim.org/</TT>'' in the example above)
must be URIs references, an XML Namespace name is <I>not required</I> to be
directly usable in the sense of identifying an actual, retrieval document
or resource on the Internet&nbsp;(<A
 HREF="sbml-level-1.html#bray:1999">Bray et&nbsp;al., 1999</A>).  The name is simply intended
to enable unique identification of constructs, and using URIs is a common
and simple way of creating a unique name string.  For the convenience of
the simulation tools developer community, we reserve certain namespace
names for use with annotations in SBML.  These reserved names are listed in
Table&nbsp;<A HREF="sbml-level-1.html#tab:reserved-urls">1</A>.

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="156"></A>
<TABLE>
<CAPTION><STRONG>Table 1:</STRONG>
Reserved XML Namespace names in SBML Level 1.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">  <TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/biospice</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/dbsolve</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/ecell</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/gepasi</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/jarnac</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/jdesigner</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/stochsim</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>http://www.sbml.org/2001/ns/vcell</TT></TD>
</TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD>
</TR>
</TABLE>
  
  <A NAME="tab:reserved-urls"></A></DIV></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>
Note that the namespaces being referred to here are XML Namespaces
specifically in the context of the <TT>annotations</TT> field on
<TT>SBase</TT>.  The namespace issue here is unrelated to the namespaces
discussed in Section&nbsp;<A HREF="sbml-level-1.html#sec:namespaces">3.4</A> below in the context of
<TT>SName</TT> and symbols in SBML.

<P>

<H2><A NAME="SECTION00033000000000000000"></A>
<A NAME="sec:name"></A><BR>
3.3 Type <TT>SName</TT>
</H2>

<P>
The type <TT>SName</TT> is used in many places in SBML for expressing names
of components in a model.  <TT>SName</TT> is is a data type derived from the
basic XML type <TT>string</TT>, but with restrictions about the types of
characters permitted and the sequence in which they may appear.  Its 
definition is shown in Figure&nbsp;<A HREF="sbml-level-1.html#fig:name">3</A>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:name"></A><A NAME="1203"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 3:</STRONG>
The definition of the type <TT>SName</TT>, expressed in conventional
    Backus-Naur Form&nbsp;(<A
 HREF="sbml-level-1.html#naur:1960">Naur, 1960</A>).  The meta symbols { and } signify
    ``zero or more times'' the items they enclose.  Note that although XML
    permits the use of Unicode characters&nbsp;(<A
 HREF="sbml-level-1.html#unicode:1996">Unicode Consortium, 1996</A>), SBML
    limits the allowable characters in <TT>SName</TT> to plain ASCII text
    characters for compatibility with existing simulation software.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">  </DIV><TABLE  WIDTH="438">
<TR><TD>
<PRE>
  letter ::= 'a'..'z','A'..'Z'
  digit  ::= '0'..'9'
  SName  ::= { '_' } letter { letter | '_' | digit }
</PRE></TD></TR>
</TABLE><DIV ALIGN="CENTER">
  </DIV></TD></TR>
</TABLE>
</DIV><P></P>

<P>
The need to define a constrained data type for names stems from the fact
that many existing simulation packages allow only a limited set of
characters in symbol names.  SBML codifies this limitation in the form of a
lowest-common-denominator data type (<TT>SName</TT>), to prevent the
creation of models with symbol names that might confuse some simulation
software packages.  This is important for facilitating model exchange
between tools.

<P>

<H2><A NAME="SECTION00034000000000000000"></A>
<A NAME="sec:namespaces"></A><BR>
3.4 Component Names and Namespaces in SBML
</H2>

<P>
A biochemical network model can contain a large number of named components
representing different parts of a model.  This leads to a problem in
deciding the scope of a symbol: in what contexts does a given symbol
<I>X</I> represent the same thing?  The approaches used in existing
simulation packages tend to fall into two categories that we may call
global and local.  The <I>global</I> approach places all symbols into a
single global namespace, so that a symbol <I>X</I> represents the same
thing wherever it appears in a given model definition.  The <I>local</I>
approach places symbols in different namespaces depending on the context,
where the context may be, for example, individual rate laws.  The latter
approach means that a user may use the same symbol <I>X</I> in different
rate laws and have each instance represent a different quantity.

<P>
The fact that different simulation programs may use different rules for
name resolution poses a problem for the exchange of models between
simulation tools.  Without careful consideration, a model written out in
SBML format by one program may be misinterpreted by another program.  SBML
must therefore include a specific set of rules for treating symbols and
namespaces.

<P>
The namespace rules in SBML Level 1 are relatively straightforward and are
intended to avoid this problem with a minimum of requirements on the
implementation of software tools:

<UL>
<LI>All model-level component names (compartments, species, reactions,
  parameters, parameter rules, and units) reside in the same global
  namespace.  This means, for example, that a reaction and a specie
  definition cannot both have the same name.

<P>
</LI>
<LI>Each reaction definition (see Section&nbsp;<A HREF="sbml-level-1.html#sec:reactions">4.7</A>)
  establishes a private local namespace for parameter names.  Within the
  definition of a given reaction, parameter names introduced in that
  reaction override (shadow) identical names in the global namespace.

<P>
</LI>
<LI>Certain names in SBML Level 1 are reserved or otherwise have special
  meaning.  Table&nbsp;<A HREF="sbml-level-1.html#tab:reserved-names">2</A> on the following page lists
  these reserved names.  They are comprised of predefined mathematical
  functions, certain operators (present and expected in the future), and
  rate law functions.  In order to prevent name collisions, these reserved
  names cannot be used as names for any component of a model.

<P>
</LI>
</UL>  

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="199"></A>
<TABLE>
<CAPTION><STRONG>Table 2:</STRONG>
The reserved names in SBML Level 1.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">  <TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><TT>abs	   </TT></TD>
<TD ALIGN="LEFT"><TT> cos	</TT></TD>
<TD ALIGN="LEFT"><TT> hillr   </TT></TD>
<TD ALIGN="LEFT"><TT> not     </TT></TD>
<TD ALIGN="LEFT"><TT> ppbr	</TT></TD>
<TD ALIGN="LEFT"><TT> tan	</TT></TD>
<TD ALIGN="LEFT"><TT> ucii </TT></TD>
<TD ALIGN="LEFT"><TT> umar  </TT></TD>
<TD ALIGN="LEFT"><TT> usii  </TT></TD>
<TD ALIGN="LEFT"><TT> volume</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    acos   </TT></TD>
<TD ALIGN="LEFT"><TT> exp	</TT></TD>
<TD ALIGN="LEFT"><TT> isouur  </TT></TD>
<TD ALIGN="LEFT"><TT> or 	    </TT></TD>
<TD ALIGN="LEFT"><TT> sin	</TT></TD>
<TD ALIGN="LEFT"><TT> umai	</TT></TD>
<TD ALIGN="LEFT"><TT> ucir </TT></TD>
<TD ALIGN="LEFT"><TT> umi   </TT></TD>
<TD ALIGN="LEFT"><TT> usir  </TT></TD>
<TD ALIGN="LEFT"><TT> xor</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    and    </TT></TD>
<TD ALIGN="LEFT"><TT> floor	</TT></TD>
<TD ALIGN="LEFT"><TT> log	  </TT></TD>
<TD ALIGN="LEFT"><TT> ordbbr  </TT></TD>
<TD ALIGN="LEFT"><TT> sqr	</TT></TD>
<TD ALIGN="LEFT"><TT> umar	</TT></TD>
<TD ALIGN="LEFT"><TT> ucti </TT></TD>
<TD ALIGN="LEFT"><TT> unii  </TT></TD>
<TD ALIGN="LEFT"><TT> uuci </TT></TD>
<TD></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    asin   </TT></TD>
<TD ALIGN="LEFT"><TT> hilli	</TT></TD>
<TD ALIGN="LEFT"><TT> log10	  </TT></TD>
<TD ALIGN="LEFT"><TT> ordbur  </TT></TD>
<TD ALIGN="LEFT"><TT> sqrt	</TT></TD>
<TD ALIGN="LEFT"><TT> uai	</TT></TD>
<TD ALIGN="LEFT"><TT> uctr </TT></TD>
<TD ALIGN="LEFT"><TT> unir  </TT></TD>
<TD ALIGN="LEFT"><TT> uucr </TT></TD>
<TD></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    atan   </TT></TD>
<TD ALIGN="LEFT"><TT> hillmmr	</TT></TD>
<TD ALIGN="LEFT"><TT> massi	  </TT></TD>
<TD ALIGN="LEFT"><TT> ordubr  </TT></TD>
<TD ALIGN="LEFT"><TT> substance	</TT></TD>
<TD ALIGN="LEFT"><TT> ualii	</TT></TD>
<TD ALIGN="LEFT"><TT> uhmi </TT></TD>
<TD ALIGN="LEFT"><TT> uuhr  </TT></TD>
<TD ALIGN="LEFT"><TT> uui </TT></TD>
<TD></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    ceil   </TT></TD>
<TD ALIGN="LEFT"><TT> hillmr	</TT></TD>
<TD ALIGN="LEFT"><TT> massr	  </TT></TD>
<TD ALIGN="LEFT"><TT> pow     </TT></TD>
<TD ALIGN="LEFT"><TT> time	</TT></TD>
<TD ALIGN="LEFT"><TT> uar	</TT></TD>
<TD ALIGN="LEFT"><TT> uhmr </TT></TD>
<TD ALIGN="LEFT"><TT> umr   </TT></TD>
<TD ALIGN="LEFT"><TT> uur </TT></TD>
<TD></TD>
</TR>
<TR><TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
</TR>
</TABLE><TT>
  
  <A NAME="tab:reserved-names"></A></TT></DIV></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>
The set of rules above can enable software packages using either local or
global namespaces to exchange SBML model definitions.  In particular,
software environments using local namespaces internally should be able to
accept SBML model definitions without needing to change component names.
Environments using a global namespace internally can perform a simple
manipulation of the names of elements within reaction definitions to avoid
name collisions.  (An example approach for the latter would be the
following: when receiving an SBML-encoded model, prefix each name inside
each reaction with a string constructed from the reaction's name; when
writing an SBML-encoded model, strip off the prefix.)

<P>
The namespace rules described here provide a clean transition path to
future levels of SBML, when submodels are introduced
(Section&nbsp;<A HREF="sbml-level-1.html#sec:level-2">6.1</A>).  Submodels will provide the ability to compose
one model from a collection of other models.  This capability will have to
be built on top of SBML Level&nbsp;1's namespace organization.  A
straightforward approach to handling namespaces is to make each submodel's
space be private.  The rules governing namespaces within a submodel can
simply be the Level&nbsp;1 namespace rule described here, with each submodel
having its own (to itself, global) namespace.

<P>

<H2><A NAME="SECTION00035000000000000000"></A>
<A NAME="sec:formulas"></A><BR>
3.5 Formulas
</H2>

<P>
Formulas in SBML Level 1 are expressed in text string form.  They are used
in the definitions of kinetic laws (Section&nbsp;<A HREF="sbml-level-1.html#subsec:kinetic-law">4.7.2</A>) and
in rules (Section&nbsp;<A HREF="sbml-level-1.html#sec:rules">4.6</A>).  The formula strings are interpreted as
expressions that evaluate to a floating-point value of type <TT>double</TT>.
The formula strings may contain operators, function calls, and symbols.
Table&nbsp;<A HREF="sbml-level-1.html#tab:operators">3</A> presents the precedence rules
for the different entities that may appear in formula strings.  All
operators in formulas return <TT>double</TT> values.

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="1299"></A>
<TABLE>
<CAPTION><STRONG>Table 3:</STRONG>
A table of the expression operators available in SBML.  In the
    <B> Class</B> column, ``operand'' implies the construct is an
    operand, ``prefix'' implies the operation is applied to the following
    arguments, ``unary'' implies there is one argument, and ``binary''
    implies there are two arguments.  The values in the
    <B> Precedence</B> column show how the order of different
    types of operation are determined.  For example, the expression <IMG
 WIDTH="61" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img1.gif"
 ALT="$ a * b + c$"> is evaluated as <!-- MATH
 $(a * b) + c$
 -->
<IMG
 WIDTH="73" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img2.gif"
 ALT="$ (a * b) + c$"> because the <TT>*</TT> operator has
    higher precedence.  The <B> Associates</B> column shows how
    the order of similar precedence operations is determined; for example,
    <IMG
 WIDTH="65" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img3.gif"
 ALT="$ a - b + c$"> is evaluated as <!-- MATH
 $(a - b) + c$
 -->
<IMG
 WIDTH="77" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img4.gif"
 ALT="$ (a - b) + c$"> because the <IMG
 WIDTH="16" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.gif"
 ALT="$ +$"> and <IMG
 WIDTH="16" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img6.gif"
 ALT="$ -$">
 operators are left-associative.  The precedence and associativity rules
    are taken from the C programming language&nbsp;(<A
 HREF="sbml-level-1.html#harbison:1995">Harbison and Steele, 1995</A>),
    except for the symbol <TT>^</TT>, which is used in C for a different
    purpose.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">
<TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><B>Tokens</B></TD>
<TH ALIGN="LEFT"><B>Operation</B></TH>
<TH ALIGN="LEFT"><B>Class</B></TH>
<TH ALIGN="CENTER"><B>Precedence</B></TH>
<TH ALIGN="LEFT"><B>Associates</B></TH>
</TR>
<TR><TD ALIGN="LEFT"><I>name</I></TD>
<TD ALIGN="LEFT">symbol reference</TD>
<TD ALIGN="LEFT">operand</TD>
<TD ALIGN="CENTER">8</TD>
<TD ALIGN="LEFT">n/a</TD>
</TR>
<TR><TD ALIGN="LEFT">(<I>expression</I>)</TD>
<TD ALIGN="LEFT">expression grouping</TD>
<TD ALIGN="LEFT">operand</TD>
<TD ALIGN="CENTER">8</TD>
<TD ALIGN="LEFT">n/a</TD>
</TR>
<TR><TD ALIGN="LEFT"><I>f</I>(<I>...</I>)</TD>
<TD ALIGN="LEFT">function call</TD>
<TD ALIGN="LEFT">prefix</TD>
<TD ALIGN="CENTER">8</TD>
<TD ALIGN="LEFT">left</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="16" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img6.gif"
 ALT="$ -$"></TD>
<TD ALIGN="LEFT">negation</TD>
<TD ALIGN="LEFT">unary</TD>
<TD ALIGN="CENTER">6</TD>
<TD ALIGN="LEFT">right</TD>
</TR>
<TR><TD ALIGN="LEFT"><code>^</code></TD>
<TD ALIGN="LEFT">power</TD>
<TD ALIGN="LEFT">binary</TD>
<TD ALIGN="CENTER">5</TD>
<TD ALIGN="LEFT">left</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="12" HEIGHT="13" ALIGN="BOTTOM" BORDER="0"
 SRC="img16.gif"
 ALT="$ *$"></TD>
<TD ALIGN="LEFT">multiplication</TD>
<TD ALIGN="LEFT">binary</TD>
<TD ALIGN="CENTER">4</TD>
<TD ALIGN="LEFT">left</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="12" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img17.gif"
 ALT="$ /$"></TD>
<TD ALIGN="LEFT">division</TD>
<TD ALIGN="LEFT">binary</TD>
<TD ALIGN="CENTER">4</TD>
<TD ALIGN="LEFT">left</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="16" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img5.gif"
 ALT="$ +$"></TD>
<TD ALIGN="LEFT">addition</TD>
<TD ALIGN="LEFT">binary</TD>
<TD ALIGN="CENTER">3</TD>
<TD ALIGN="LEFT">left</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="16" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img6.gif"
 ALT="$ -$"></TD>
<TD ALIGN="LEFT">subtraction</TD>
<TD ALIGN="LEFT">binary</TD>
<TD ALIGN="CENTER">3</TD>
<TD ALIGN="LEFT">left</TD>
</TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="CENTER">&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD>
</TR>
</TABLE>
</DIV>
  
  <A NAME="tab:operators"></A></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>
The function call syntax consists of a function name, followed by optional
white space, followed by an opening parenthesis token (`('), followed by a
sequence of zero or more arguments separated by commas, followed by a
closing parenthesis (`)') token.  The function name must be chosen from one
of the functions available in SBML.  Table&nbsp;<A HREF="sbml-level-1.html#tab:simplemath">6</A> in
Appendix&nbsp;<A HREF="sbml-level-1.html#apdx:predefined-functions">C</A> lists the basic mathematical
functions that are defined in SBML at this time, while
Table&nbsp;<A HREF="sbml-level-1.html#tab:ratelaws">7</A> lists a large number of common rate law functions
defined in SBML.  The names of these predefined functions are reserved and
make up the bulk of the list of names in
Table&nbsp;<A HREF="sbml-level-1.html#tab:reserved-names">2</A>.

<P>
A program parsing a formula in an SBML model should assume that name tokens
other than function names are names of parameters, parameter rules,
compartments or species.  When a specie name occurs in a formula, it
represents the concentration (i.e., <!-- MATH
 $substance/volume$
 -->
<IMG
 WIDTH="132" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img18.gif"
 ALT="$ substance/volume$">) of the specie.
When a compartment name occurs in a formula, it represents the volume of
the compartment.  The units of substance and volume are determined from the
built-in <TT>substance</TT> and <TT>volume</TT> of Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>.

<P>
Readers may wonder why mathematical formulas in SBML are not expressed
using MathML&nbsp;(<A
 HREF="sbml-level-1.html#w3c:2000b">W3C, 2000b</A>), an XML-based mathematical formula language.
Although using MathML would be more in the spirit of using XML and would in
some ways be a more forward-looking choice, it would require simulation
software to use fairly complex parsers to read and write the resulting
SBML.  Most contemporary systems biology simulation software simply
represent mathematical formulas using text strings.  To keep SBML Level&nbsp;1
simple and compatible with known simulation software, we chose to represent
formulas as strings.  This does not preclude a later level of SBML from
introducing the ability to use MathML as an extension.

<P>

<H1><A NAME="SECTION00040000000000000000"></A>
<A NAME="sec:elements"></A><BR>
4 SBML Components
</H1>

<P>
In this section, we define each of the major data structures in SBML.  To
provide illustrations of their use, we give partial XML encodings of SBML
model components, but we leave full XML examples to
Section&nbsp;<A HREF="sbml-level-1.html#sec:xml-rep">5</A>.

<P>

<H2><A NAME="SECTION00041000000000000000"></A>
<A NAME="sec:model"></A><BR>
4.1 Models
</H2>

<P>
The <TT>Model</TT> structure is the highest-level construct in an SBML data
stream or document.  It defines a grouping of components--the list of
compartments, species, reactions, parameters, rules and unit definitions
that define a given model.  Only one component of type <TT>Model</TT> is
allowed per instance of an SBML document or data stream, although it does
not necessarily need to represent a single biological entity.  The UML
definition of the <TT>Model</TT> structure is shown in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:model">4</A>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:model"></A><A NAME="1207"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 4:</STRONG>
The definition of <TT>Model</TT>.  Additional fields are
    inherited from <TT>SBase</TT>.</CAPTION>
<TR><TD><IMG
 WIDTH="216" HEIGHT="140" BORDER="0"
 SRC="img19.gif"
 ALT="\begin{figure}\ \centering \incfig[scale = 0.68]{model}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
A model must contain at least one <TT>Specie</TT>, one <TT>Reaction</TT> and
one <TT>Compartment</TT> data element, but SBML does not impose restrictions
on the total number of these.  A <TT>Model</TT> data object may optionally
have lists of <TT>UnitDefinition</TT>, <TT>Parameter</TT> and <TT>Rule</TT>
components (they are optional because the lists in fields
<TT>unitDefinition</TT>, <TT>compartment</TT>, and <TT>rule</TT>
are permitted to have zero length).  It may also have an optional
<TT>name</TT> field that can be used to give the model a name.  The name
must be a text string conforming to the syntax permitted by the
<TT>SName</TT> data type described in Section&nbsp;<A HREF="sbml-level-1.html#sec:name">3.3</A>.

<P>
In the XML encoding of an SBML model, the lists of species, reactions,
compartments, and optional unit definitions, parameters and rules, are
translated into lists of XML elements that each have headings of the form
<TT>listOfs</TT>, where the blank is replaced by
the name of the component type (e.g., ``<TT>Reaction</TT>'').  The
resulting XML data object has the form illustrated by the following
skeletal model:

<P>
<BR>
<IMG
 WIDTH="209" HEIGHT="293" ALIGN="BOTTOM" BORDER="0"
 SRC="img20.gif"
 ALT="\begin{example}
&lt;model name=''My_Model''&gt;
&lt;listOfUnitDefinitions&gt;
...
&lt;/listO...
.../listOfRules&gt;
&lt;listOfReactions&gt;
...
&lt;/listOfReactions&gt;
&lt;/model&gt;
\end{example}">
<BR>

<P>
Readers may wonder about the motivations for the
<TT>listOfs</TT> notation.  A simpler approach to
creating the lists of components would be to place them all directly
at the top level under <TT>&lt;model&gt; ... &lt;/model&gt;</TT>.  We chose instead to
group them within XML elements named after
<TT>listOfs</TT>, because we believe this helps
organize the components and makes visual reading of model definitions
easier.

<P>

<H2><A NAME="SECTION00042000000000000000"></A>
<A NAME="sec:unitdefinitions"></A><BR>
4.2 Unit Definitions
</H2>

<P>
Units may be supplied in a number of contexts in an SBML model.  A facility
for defining units is convenient to have so that combinations of units can
be given abbreviated names.  This is the motivation behind the
<TT>UnitDefinition</TT> data structure, whose definition is shown in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:unitdefinition">5</A>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:unitdefinition"></A><A NAME="1208"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 5:</STRONG>
The definition of <TT>UnitDefinition</TT>.</CAPTION>
<TR><TD><IMG
 WIDTH="475" HEIGHT="78" BORDER="0"
 SRC="img21.gif"
 ALT="\begin{figure}\centering \incfig[scale = 0.68]{unitdefinition}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
A unit definition consists of a <TT>name</TT> field of type <TT>SName</TT>
and an optional list of structures of type <TT>Unit</TT>.  The approach to
defining units in SBML is compositional; for example, <!-- MATH
 $meter\
second^{\,-2}$
 -->
<IMG
 WIDTH="118" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img22.gif"
 ALT="$ meter\
second^{\,-2}$"> is constructed by combining a <TT>Unit</TT>-type element
representing <IMG
 WIDTH="46" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img23.gif"
 ALT="$ meter$"> with a <TT>Unit</TT>-type element representing
<!-- MATH
 $second^{\,-2}$
 -->
<IMG
 WIDTH="71" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img24.gif"
 ALT="$ second^{\,-2}$">.  The <TT>Unit</TT> data structure has a <TT>kind</TT> field
whose value must be taken from <TT>UnitKind</TT>, an enumeration of SI
units.  The possible values of <TT>UnitKind</TT> are listed in
Table&nbsp;<A HREF="sbml-level-1.html#tab:unitkind">4</A>.  The <TT>exponent</TT> field on <TT>Unit</TT>
represents an exponent on the unit.  Its default value is
``<TT>1</TT>'' (one).  In the example just mentioned, <!-- MATH
 $second^{\,-2}$
 -->
<IMG
 WIDTH="71" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img24.gif"
 ALT="$ second^{\,-2}$">
is obtained by using <TT>kind="second"</TT> and <TT>exponent="-2"</TT>.
Finally, the <TT>scale</TT> field in <TT>Unit</TT> is an integer attribute
that scales the unit.  For example, a unit that has a <TT>kind</TT> value
of ``<TT>gram</TT>'' and a <TT>scale</TT> value of ``<TT><IMG
 WIDTH="24" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img25.gif"
 ALT="$ -3$"></TT>''
signifies <!-- MATH
 $10^{-3} * gram$
 -->
<IMG
 WIDTH="90" HEIGHT="33" ALIGN="MIDDLE" BORDER="0"
 SRC="img26.gif"
 ALT="$ 10^{-3} * gram$">, or milligrams.

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="1209"></A>
<TABLE>
<CAPTION><STRONG>Table 4:</STRONG>
The possible values of <TT>kind</TT> in a <TT>UnitKind</TT>
    structure.  All are names of base or derived SI units, except for
    ``<TT>dimensionless</TT>'' and ``<TT>item</TT>'', which are 
    SBML additions important for handling certain common cases.
    ``<TT>Dimensionless</TT>'' is intended for cases where a quantity does not
    have units, and ``<TT>item</TT>'' is  needed in certain contexts to express
    such things as ``N items'' (e.g., ``100 molecules'').
    Strictly speaking, ``<TT>celsius</TT>'' should be capitalized; however,
    for simplicity, SBML requires that the values of <TT>UnitKind</TT> be
    treated in a case-insensitive manner by software reading and writing SBML.
    Also, note that the gram and liter/litre are not
    strictly part of SI&nbsp;(<A
 HREF="sbml-level-1.html#taylor:1995">Taylor, 1995</A>); however, they are so
    useful in SBML's areas of application that they are included in the
    <TT>UnitKind</TT> enumeration of unit names.  (The standard SI unit of
    mass is in fact the kilogram, and volume is
    defined in terms of cubic meters.)</CAPTION>
<TR><TD> <DIV ALIGN="CENTER"><TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><TT> ampere      </TT></TD>
<TD ALIGN="LEFT"><TT> farad	</TT></TD>
<TD ALIGN="LEFT"><TT> joule		</TT></TD>
<TD ALIGN="LEFT"><TT> lumen		</TT></TD>
<TD ALIGN="LEFT"><TT> ohm     </TT></TD>
<TD ALIGN="LEFT"><TT> steradian</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    becquerel   </TT></TD>
<TD ALIGN="LEFT"><TT> gram	</TT></TD>
<TD ALIGN="LEFT"><TT> katal		</TT></TD>
<TD ALIGN="LEFT"><TT> lux		</TT></TD>
<TD ALIGN="LEFT"><TT> pascal  </TT></TD>
<TD ALIGN="LEFT"><TT> tesla</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    candela	</TT></TD>
<TD ALIGN="LEFT"><TT> gray	</TT></TD>
<TD ALIGN="LEFT"><TT> kelvin	</TT></TD>
<TD ALIGN="LEFT"><TT> meter		</TT></TD>
<TD ALIGN="LEFT"><TT> radian  </TT></TD>
<TD ALIGN="LEFT"><TT> volt</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    celsius     </TT></TD>
<TD ALIGN="LEFT"><TT> henry	</TT></TD>
<TD ALIGN="LEFT"><TT> kilogram	</TT></TD>
<TD ALIGN="LEFT"><TT> metre		</TT></TD>
<TD ALIGN="LEFT"><TT> second  </TT></TD>
<TD ALIGN="LEFT"><TT> watt</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    coulomb	</TT></TD>
<TD ALIGN="LEFT"><TT> hertz	</TT></TD>
<TD ALIGN="LEFT"><TT> liter		</TT></TD>
<TD ALIGN="LEFT"><TT> mole		</TT></TD>
<TD ALIGN="LEFT"><TT> siemens </TT></TD>
<TD ALIGN="LEFT"><TT> weber</TT></TD>
</TR>
<TR><TD ALIGN="LEFT"><TT> 
    <U>dimensionless</U> </TT></TD>
<TD ALIGN="LEFT"><TT> <U>item</U> </TT></TD>
<TD ALIGN="LEFT"><TT> litre	</TT></TD>
<TD ALIGN="LEFT"><TT> newton	</TT></TD>
<TD ALIGN="LEFT"><TT> sievert</TT></TD>
<TD></TD>
</TR>
<TR><TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
<TD></TD>
</TR>
</TABLE><TT>
  
  <A NAME="tab:unitkind"></A></TT></DIV></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>
Unit combinations are constructed by listing several <TT>Unit</TT>
structures inside a <TT>UnitDefinition</TT>-type structure.  The following
example illustrates the definition of an abbreviation named
``<TT>mmls</TT>'' for the units <!-- MATH
 $mmol\ l^{-1}\ s^{-1}$
 -->
<IMG
 WIDTH="102" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img27.gif"
 ALT="$ mmol\ l^{-1}\ s^{-1}$">:

<P>
<BR>
<IMG
 WIDTH="352" HEIGHT="130" ALIGN="BOTTOM" BORDER="0"
 SRC="img28.gif"
 ALT="\begin{example}
&lt;listOfUnitDefinitions&gt;
\ &lt;unitDefinition name=''mmls''&gt;
&lt;listO...
...1''/&gt;
&lt;/listOfUnits&gt;
&lt;/unitDefinition&gt;
&lt;/listOfUnitDefinitions&gt;
\end{example}">
<BR>

<P>
Many of the components in a model refer to quantities that have associated
units.  SMBL Level&nbsp;1 has three predefined quantity types: amount of
substance, time, and volume.  SBML defines default units and scales for
these quantities.  The defaults are summarized in Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>.

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="358"></A>
<TABLE>
<CAPTION><STRONG>Table 5:</STRONG>
SBML's built-in quantities and their default
    scale values.  The names in the left-hand column are reserved.  These
    names may be used wherever
    units may be supplied in a model component.</CAPTION>
<TR><TD><DIV ALIGN="CENTER">  <TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><B>Name</B></TD>
<TH ALIGN="LEFT"><B>Allowable Units</B></TH>
<TH ALIGN="LEFT"><B>Default Units</B></TH>
<TH ALIGN="CENTER"><B>Default Scale</B></TH>
</TR>
<TR><TD ALIGN="LEFT"><TT>substance</TT></TD>
<TD ALIGN="LEFT">moles or no. of molecules</TD>
<TD ALIGN="LEFT">moles</TD>
<TD ALIGN="CENTER">1</TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>time</TT></TD>
<TD ALIGN="LEFT">seconds</TD>
<TD ALIGN="LEFT">seconds</TD>
<TD ALIGN="CENTER">1</TD>
</TR>
<TR><TD ALIGN="LEFT"><TT>volume</TT></TD>
<TD ALIGN="LEFT">liters</TD>
<TD ALIGN="LEFT">liters</TD>
<TD ALIGN="CENTER">1</TD>
</TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="CENTER">&nbsp;</TD>
</TR>
</TABLE>
  
  <A NAME="tab:builtin"></A></DIV></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>
Wherever unit specifications are permitted in a model (for example, for the
volume in a compartment), the relevant built-in name from
Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A> may be used.  Such usage signifies that the units
to be used for the quantity should be the designated defaults.  A model may
change the default scales by reassigning the keywords
``<TT>substance</TT>'', ``<TT>time</TT>'', and ``<TT>volume</TT>'' in a
unit definition.  This takes advantage of the <TT>UnitDefinition</TT>
structure's facility for defining scales on units.  The following example
changes the default units of volume to be milliliters:

<P>
<BR>
<IMG
 WIDTH="360" HEIGHT="159" ALIGN="BOTTOM" BORDER="0"
 SRC="img29.gif"
 ALT="\begin{example}
&lt;model&gt;
...
&lt;listOfUnitDefinitions&gt;
&lt;unitDefinition name=''vo...
...Units&gt;
&lt;/unitDefinition&gt;
&lt;/listOfUnitDefinitions&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>
If the definition above appeared in a model, the volume scale on all
components that did not explicitly use different units would be changed to
milliliters.  

<P>
The list of unit definitions in a <TT>Model</TT>-type structure is the only
place where new units can be defined.  The new unit names may be used
anywhere in a model where unit specifications are permitted.  The various
components of a model, such as reaction parameters and rules, can only use
the base units from Table&nbsp;<A HREF="sbml-level-1.html#tab:unitkind">4</A>, the global unit definitions
in the model, or the three predefined keywords ``<TT>substance</TT>'',
``<TT>time</TT>'', and ``<TT>volume</TT>''.

<P>

<H2><A NAME="SECTION00043000000000000000"></A>
<A NAME="sec:compartments"></A><BR>
4.3 Compartments
</H2>

<P>
A <TT>Compartment</TT> represents a bounded container in which species are
located.  The definition of <TT>Compartment</TT> is shown in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:compartment">6</A>.  

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:compartment"></A><A NAME="1210"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 6:</STRONG>
The definition of <TT>Compartment</TT>.  
    Fields inherited from <TT>SBase</TT> are omitted here but are assumed.</CAPTION>
<TR><TD><IMG
 WIDTH="279" HEIGHT="94" BORDER="0"
 SRC="img30.gif"
 ALT="\begin{figure}\vspace*{8pt}
\centering \incfig[scale = 0.68]{compartment}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
A <TT>Compartment</TT> data object has a <TT>name</TT> field of type
<TT>SName</TT>.  A compartment also has a floating-point field called
<TT>volume</TT>, representing the total volume of the compartment in the
default units of volume.  (See Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>.)  This enables
concentrations of species to be calculated in the absence of geometry
information.  The <TT>volume</TT> field is optional and defaults to a value
of ``<TT>1</TT>'' (one).

<P>
The units of volume may be explicitly set using the optional field
<TT>units</TT> in <TT>Compartment</TT>; the named units must be either one
of the base units from Table&nbsp;<A HREF="sbml-level-1.html#tab:unitkind">4</A>, the built-in default
named <TT>volume</TT>, or a new unit defined by a unit definition in the
enclosing model.  If absent, the units default to the value set by the
built-in <TT>volume</TT> of Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>.

<P>
The optional field <TT>outside</TT> of type <TT>SName</TT> can be used to
express containment relationships between compartments.  If present, the
value of <TT>outside</TT> for a given compartment should be the name of the
compartment enclosing it, or in other words, the compartment that is
``outside'' of it.  This facility can be used to model cell membranes.  For
example, to express that a compartment named B has a membrane that is
modeled as another compartment M, which in turn is located within another
compartment A, one would write:
<BR>
<IMG
 WIDTH="322" HEIGHT="130" ALIGN="BOTTOM" BORDER="0"
 SRC="img31.gif"
 ALT="\begin{example}
&lt;model&gt;
...
&lt;listOfCompartments&gt;
&lt;compartment name=''A''/&gt;
&lt;...
...nt name=''B'' outside=''M''/&gt;
&lt;/listOfCompartments&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>
In the absence of a value for <TT>outside</TT>, compartment definitions in
SBML Level&nbsp;1 do not have any implied spatial relationships between each
other.  Thus, compartments may be adjacent to each other or have other
spatial relationships.  For many modeling applications, the transfer of
substances described by the reactions in a model sufficiently express the
relationships between the compartments.  (SBML Level&nbsp;1 currently does not
provide for spatial characteristics aside from compartment volume and
containment.  As discussed in Section&nbsp;<A HREF="sbml-level-1.html#sec:level-2">6.1</A>, we expect that
SBML Level&nbsp;2 will introduce the ability to define geometries and spatial
qualities.)

<P>
In an XML data stream containing an SBML model, compartments are listed
inside an XML element called <TT>listOfCompartments</TT> within a
<TT>Model</TT>-type data structure.  (See the discussion of <TT>Model</TT> in
Section&nbsp;<A HREF="sbml-level-1.html#sec:model">4.1</A>.)  The following example illustrates two
compartments in an abbreviated SBML example of a model definition:

<P>
<BR>
<IMG
 WIDTH="412" HEIGHT="115" ALIGN="BOTTOM" BORDER="0"
 SRC="img32.gif"
 ALT="\begin{example}
&lt;model&gt;
...
&lt;listOfCompartments&gt;
&lt;compartment name=''cytosol'...
...tochondria'' volume=''0.3''/&gt;
&lt;/listOfCompartments&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>

<H2><A NAME="SECTION00044000000000000000"></A>
<A NAME="sec:species"></A><BR>
4.4 Species
</H2>

<P>
The term <I>species</I> refers to entities that take part in reactions.
These include simple ions (e.g., protons, calcium), simple molecules (e.g.,
glucose, ATP), and large molecules (e.g., RNA, polysaccharides, and
proteins).  The <TT>Specie</TT> data structure is intended to represent
these entities.  Its definition is shown in Figure&nbsp;<A HREF="sbml-level-1.html#fig:specie">7</A>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:specie"></A><A NAME="1211"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 7:</STRONG>
The definition of <TT>Specie</TT>.  As usual, fields inherited from
    <TT>SBase</TT> are omitted here but are assumed.</CAPTION>
<TR><TD><IMG
 WIDTH="400" HEIGHT="124" BORDER="0"
 SRC="img33.gif"
 ALT="\begin{figure}\centering \incfig[scale = 0.68]{specie}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
<TT>Specie</TT> has a <TT>name</TT> field of type <TT>SName</TT>.  The field
<TT>compartment</TT>, also of type <TT>SName</TT>, is used to identify the
compartment in which the specie is located.  The field
<TT>initialAmount</TT>, of type <TT>double</TT>, is used to set the initial
amount of the specie in the named compartment.  The units of the substance
quantity may be explicitly set using the optional field <TT>units</TT>.
The value assigned to <TT>units</TT> must be chosen from one of the
following possibilities: one of base unit names from
Table&nbsp;<A HREF="sbml-level-1.html#tab:unitkind">4</A>, the name ``<TT>volume</TT>'', or a new unit name
defined by a unit definition in the enclosing model.  If absent, the units
default to the value set by the built-in ``<TT>substance</TT>'' of
Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>.

<P>
The optional boolean field <TT>boundaryCondition</TT> determines whether
the amount of the specie is fixed or variable over the course of a
simulation.  <TT>boundaryCondition</TT> defaults to a value of
``<TT>false</TT>'', indicating that by default, the amount is not fixed.
The optional field <TT>charge</TT> is an integer indicating the charge on
the species (in terms of electrons, not the SI unit Coulombs).  This may be
useful when the specie involved is a charged ion such as calcium
(<!-- MATH
 $\text{Ca}^{2++}$
 -->
Ca<IMG
 WIDTH="31" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img12.gif"
 ALT="$ ^{2++}$">).

<P>
The following example shows two specie definitions within an abbreviated
SBML model definition.  The example shows that species are listed under the
heading <TT>listOfSpecies</TT> in the model:

<P>
<BR>
<IMG
 WIDTH="570" HEIGHT="115" ALIGN="BOTTOM" BORDER="0"
 SRC="img34.gif"
 ALT="\begin{example}
&lt;model&gt;
\ ...
&lt;listOfSpecies&gt;
&lt;specie name=''Glucose'' compart...
...ell'' initialAmount=''0.75''/&gt;
&lt;/listOfSpecies&gt;
...
&lt;/model&gt;
\par\end{example}">
<BR>

<P>
Note that the compartment's name is used instead of an identifier, because
simulation systems typically enable users to locate species using
compartment names rather than machine-generated identifiers.  Since a
compartment's name must be unique among all the compartments in a model
(see the discussion of namespaces in Section&nbsp;<A HREF="sbml-level-1.html#sec:namespaces">3.4</A>), there
is no danger of ambiguity and no compelling reason to introduce identifiers
on compartments.

<P>

<H2><A NAME="SECTION00045000000000000000"></A>
<A NAME="sec:parameters"></A><BR>
4.5 Parameters
</H2>

<P>
A <TT>Parameter</TT> structure is used to associate a symbol with a
floating-point value so that the symbol can be used in formulas in place of
the value.  The definition of <TT>Parameter</TT> is shown in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:parameter">8</A>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:parameter"></A><A NAME="1212"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 8:</STRONG>
The definition of <TT>Parameter</TT>.</CAPTION>
<TR><TD><IMG
 WIDTH="210" HEIGHT="76" BORDER="0"
 SRC="img35.gif"
 ALT="\begin{figure}\centering \incfig[scale = 0.68]{parameter}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
<TT>Parameter</TT> has a <TT>name</TT> field of type <TT>SName</TT>.  This
name represents the symbol.  The field <TT>value</TT> determines the value
(of type <TT>double</TT>) assigned to the symbol.  The units of the
parameter <TT>value</TT> are specified by the field <TT>units</TT>.  The
value assigned to <TT>units</TT> must be chosen from one of the following
possibilities: one of base unit names from Table&nbsp;<A HREF="sbml-level-1.html#tab:unitkind">4</A>; one of
the three names ``<TT>substance</TT>'', ``<TT>time</TT>'', or
``<TT>volume</TT>'' (see Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>); or the name of a new
unit defined in the list of unit definitions in the enclosing <TT>Model</TT>
structure.

<P>
Parameters are used in two places in SBML: in lists of parameters defined
at the top level in a <TT>Model</TT>-type structure, and within individual
reaction definitions.  Parameters defined at the top level are
<I>global</I> to the whole model; parameters that are defined within
a reaction are local to the particular reaction and (within that reaction)
<I>override</I> any global parameters having the same names.  (See
Section&nbsp;<A HREF="sbml-level-1.html#sec:namespaces">3.4</A> for further details.)

<P>
The following is an example of parameters defined at the <TT>Model</TT> level:

<P>
<BR>
<IMG
 WIDTH="442" HEIGHT="204" ALIGN="BOTTOM" BORDER="0"
 SRC="img36.gif"
 ALT="\begin{example}
&lt;model&gt;
...
&lt;listOfSpecies&gt;
...
&lt;/listOfSpecies&gt;
&lt;listOfPar...
...ameters&gt;
&lt;listOfReactions&gt;
...
&lt;/listOfReactions&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>
An example of a full model that uses parameters is presented in
Section&nbsp;<A HREF="sbml-level-1.html#subsection:ruleseg">5.3</A>.

<P>

<H2><A NAME="SECTION00046000000000000000"></A>
<A NAME="sec:rules"></A><BR>
4.6 Rules
</H2>

<P>
In SBML, <I>rules</I> provide a way to create constraints on variables for
cases in which the constraints cannot be expressed using the reaction
components (Section&nbsp;<A HREF="sbml-level-1.html#sec:reactions">4.7</A>).  There are two orthogonal
dimensions by which rules can be described.  First, there are three
different possible functional forms, corresponding to the following three
general cases (where <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> and <IMG
 WIDTH="12" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img38.gif"
 ALT="$ y$"> are variables, <IMG
 WIDTH="14" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img39.gif"
 ALT="$ f$"> is some arbitrary
function, and <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"> is a vector of parameters):

<P>
<DIV ALIGN="CENTER">
<TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT">Case 1, left-hand side is zero:</TD>
<TD ALIGN="LEFT"><!-- MATH
 $0 = f(x, P)$
 -->
<IMG
 WIDTH="84" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img41.gif"
 ALT="$ 0 = f(x, P)$"></TD>
</TR>
<TR><TD ALIGN="LEFT">Case 2, left-hand side is a scalar:</TD>
<TD ALIGN="LEFT"><IMG
 WIDTH="69" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img42.gif"
 ALT="$ x = f(P)$"></TD>
</TR>
<TR><TD ALIGN="LEFT">Case 3, left-hand side is a rate-of-change:</TD>
<TD ALIGN="LEFT"><!-- MATH
 $dx/dt = f(x, P)$
 -->
<IMG
 WIDTH="115" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img43.gif"
 ALT="$ dx/dt = f(x, P)$"></TD>
</TR>
</TABLE>
</DIV>

<P>
The second dimension concerns the role of the variable <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> in the equations
above: <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> can be the name of a compartment (to set its volume), the name
of a specie (to set its concentration), or a parameter name (to set its
value).

<P>
The approach taken to covering these cases in SBML is to define an abstract
<TT>Rule</TT> structure that contains just one field, <TT>formula</TT>, to
hold the right-hand side expression, then to derive subtypes of
<TT>Rule</TT> that add fields to cover the various cases above.
Figure&nbsp;<A HREF="sbml-level-1.html#fig:rules">9</A> gives the definitions of <TT>Rule</TT> and the
subtypes derived from it.  The figure shows that <TT>AlgebraicRule</TT> is
defined directly from <TT>Rule</TT>, whereas <TT>CompartmentVolumeRule</TT>,
<TT>SpecieConcentrationRule</TT>, and <TT>ParameterRule</TT> are all derived
from an intermediate abstract structure called <TT>AssignmentRule</TT>.

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:rules"></A><A NAME="1213"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 9:</STRONG>
The definition of <TT>Rule</TT> and derived types.</CAPTION>
<TR><TD><IMG
 WIDTH="624" HEIGHT="222" BORDER="0"
 SRC="img44.gif"
 ALT="\begin{figure}\ \centering \incfig[scale = 0.68]{rule}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
The <TT>type</TT> field introduced in <TT>AssignmentRule</TT> is an
enumeration of type <TT>RuleType</TT> that determines whether a rule
falls into category 2 or 3 in the list of cases above.  In SBML Level&nbsp;1,
the enumeration has two possible values: ``<TT>scalar</TT>'' and
``<TT>rate</TT>''.  The former means that the expression has a scalar value
on the left-hand side [i.e., <!-- MATH
 $x = f(x, P)$
 -->
<IMG
 WIDTH="85" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img45.gif"
 ALT="$ x = f(x, P)$">, as in case 2 in the list
above]; the latter means that the expression has a rate of change
differential on the left-hand side (as in case 3 in the list above).
Future releases of SBML may add to the possible values of
<TT>RuleType</TT>.

<P>

<H3><A NAME="SECTION00046100000000000000">
4.6.1 <TT>AlgebraicRule</TT></A>
</H3>

<P>
The rule type <TT>AlgebraicRule</TT> is used to express equations whose
left-hand sides are zero.  <TT>AlgebraicRule</TT> does not add any fields to
the basic <TT>Rule</TT>; its role is simply to distinguish this case from
the other cases.

<P>

<H3><A NAME="SECTION00046200000000000000">
4.6.2 <TT>SpeciesConcentrationRule</TT></A>
</H3>

<P>
The <TT>SpeciesConcentrationRule</TT> structure adds one field,
<TT>specie</TT>, to the basic <TT>AssignmentRule</TT> type.  The field
<TT>specie</TT> has type <TT>SName</TT> and is used to identify the specie
affected by the rule.  The effect of the rule depends on the value of
<TT>type</TT>: if the value is ``<TT>scalar</TT>'', the rule sets the
referenced specie's concentration to the value determined by the formula;
if the value is ``<TT>rate</TT>'', the rule sets the rate of change of the
specie's concentration to the value determined by the formula.  The units
are in terms of <!-- MATH
 $substance/volume$
 -->
<IMG
 WIDTH="132" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img18.gif"
 ALT="$ substance/volume$">, where the <IMG
 WIDTH="73" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img46.gif"
 ALT="$ substance$"> units are those
that are declared on the referenced <TT>Specie</TT> element, and the
<IMG
 WIDTH="56" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img47.gif"
 ALT="$ volume$"> units are those declared on the <TT>compartment</TT> element that
contains the <TT>Specie</TT>.

<P>

<H3><A NAME="SECTION00046300000000000000">
4.6.3 <TT>CompartmentVolumeRule</TT></A>
</H3>

<P>
The <TT>CompartmentRule</TT> structure adds one field, <TT>compartment</TT>,
to the basic <TT>AssignmentRule</TT> type.  The field <TT>compartment</TT>
has type <TT>SName</TT> and is used to identify the compartment affected by
the assignment.  The effect of the rule depends on the value of
<TT>type</TT>: if the type is ``<TT>scalar</TT>'', the rule sets the
referenced compartment's volume to the volume determined by the formula; if
the type is ``<TT>rate</TT>'', the rule sets the rate of change of the
compartment's volume to the volume determined by the formula.

<P>

<H3><A NAME="SECTION00046400000000000000">
4.6.4 <TT>ParameterRule</TT></A>
</H3>

<P>
The <TT>ParameterRule</TT> structure adds two fields, <TT>name</TT> and
<TT>units</TT>, to the basic <TT>AssignmentRule</TT> type.  The
<TT>name</TT> attribute has type <TT>SName</TT> and identifies the
parameter.  The <TT>units</TT> field acts in the same way as in the case of
the <TT>Parameter</TT> structure (Section&nbsp;<A HREF="sbml-level-1.html#sec:parameters">4.5</A>).  The value
assigned to <TT>units</TT> must be chosen from one of the following
possibilities: one of base unit names from table&nbsp;<A HREF="sbml-level-1.html#tab:unitkind">4</A>; one of
the three names ``<TT>substance</TT>'', ``<TT>time</TT>'', or
``<TT>volume</TT>'' (see Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>); or the name of a new
unit defined in the list of unit definitions in the enclosing <TT>model</TT>
structure.  The effect of this rule is to give a value to a parameter that
can be used in subsequent formulas.  This parameter has the value returned
by the expression in the <TT>formula</TT> attribute.

<P>

<H3><A NAME="SECTION00046500000000000000">
4.6.5 Guidelines for Evaluating Rules</A>
</H3>

<P>
Rules of type ``<TT>scalar</TT>'' should be evaluated first in the order
given in the SBML model definition, followed by the rules of type
``<TT>rate</TT>'' in the order given.  Following this, the ``<TT>rate</TT>''
type rules can be evaluated any number of times before the
``<TT>scalar</TT>'' type rules are re-evaluated.

<P>

<H3><A NAME="SECTION00046600000000000000">
4.6.6 Example of Rule Use</A>
</H3>

<P>
The following is an example use of rules:

<P>
<BR>
<IMG
 WIDTH="525" HEIGHT="130" ALIGN="BOTTOM" BORDER="0"
 SRC="img48.gif"
 ALT="\begin{example}
&lt;model&gt;
\ ...
\ &lt;listOfRules&gt;
&lt;parameterRule name=''k'' formula...
...rtment=''A'' formula=''0.10 * t''/&gt;
&lt;/listOfRules&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>

<H2><A NAME="SECTION00047000000000000000"></A>
<A NAME="sec:reactions"></A><BR>
4.7 Reactions
</H2>

<P>
A <I>reaction</I> represents some transformation, transport or binding
process, typically a chemical reaction, that can change the amount of one
or more species.  The <TT>Reaction</TT> type is defined in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:reaction">10</A>.  

<P>

<P></P>
<DIV ALIGN="CENTER"><A NAME="fig:reaction"></A><A NAME="1218"></A>
<TABLE>
<CAPTION ALIGN="BOTTOM"><STRONG>Figure 10:</STRONG>
The definitions of <TT>Reaction</TT>, <TT>KineticLaw</TT> and
    <TT>SpecieReference</TT>.</CAPTION>
<TR><TD><IMG
 WIDTH="616" HEIGHT="241" BORDER="0"
 SRC="img49.gif"
 ALT="\begin{figure}\centering \incfig[scale = 0.68]{reaction}
\end{figure}"></TD></TR>
</TABLE>
</DIV><P></P>

<P>
In SBML, reactions are defined using lists of reactant species, products,
and their stoichiometries, and by parameter values for separately-defined
kinetic laws.  These various quantities are recorded in the fields
<TT>reactant</TT>, <TT>product</TT>, and <TT>kineticLaw</TT>.  Both
<TT>reactant</TT> and <TT>product</TT> are references to species
implemented using lists of <TT>SpecieReference</TT> structures (defined in
Section&nbsp;<A HREF="sbml-level-1.html#subsec:speciesreference">4.7.1</A> below).  The <TT>SpecieReference</TT>
structure contains fields for recording the names of species and their
stoichiometries.  <TT>kineticLaw</TT> is an optional field of type
<TT>KineticLaw</TT> (defined in Section&nbsp;<A HREF="sbml-level-1.html#subsec:kinetic-law">4.7.2</A> below),
used to provide a mathematical formula describing the rate of the reaction.

<P>
In addition to these fields, the <TT>Reaction</TT> structure also has a
boolean field, <TT>reversible</TT>, that indicates whether the reaction is
reversible.  The field is optional, and if left unspecified in a model, it
defaults to a value of ``<TT>true</TT>''.  Information about
reversibility is useful in certain kinds of structural analyses such as
elementary mode analysis.

<P>
The field <TT>fast</TT> is another boolean attribute in the
<TT>Reaction</TT> data structure; a value of ``<TT>true</TT>''
signifies that the given reaction is a ``fast'' one.  This may be relevant
when computing equilibrium concentrations of rapidly equilibrating
reactions.  Simulation/analysis packages may chose to use this information
to reduce the number of ODEs required and thereby optimize such
computations.  The default value of <TT>fast</TT> is
``<TT>false</TT>''.  (A simulator/analysis package that has no
facilities for dealing with fast reactions can ignore this attribute.  In
theory, if the choice of which reactions are fast is correctly made, then a
simulation performed with them should give the same results as a simulation
performed without fast reactions.  However, currently there appears to be
no single unambiguous method for designating which reactions should be
considered fast, and some users may designate a reaction as fast when in
fact it is not.  Caveat developer.)

<P>

<H3><A NAME="SECTION00047100000000000000"></A>
<A NAME="subsec:speciesreference"></A><BR>
4.7.1 <TT>SpecieReference</TT>
</H3>

<P>
Each unique specie involved in a reaction is listed once in a model, in a
list contained in the <TT>specie</TT> field of the <TT>Model</TT> data
structure discussed in Section&nbsp;<A HREF="sbml-level-1.html#sec:model">4.1</A>.  Lists of products and
reactants in <TT>Reaction</TT> type structures refer to those species.  The
connection between the products and reactants in a reaction definition and
the specie names listed in the enclosing <TT>Model</TT> definition is
achieved using the <TT>SpecieReference</TT> type data structure defined in
Figure&nbsp;<A HREF="sbml-level-1.html#fig:reaction">10</A>.

<P>
The field <TT>specie</TT> of type <TT>SName</TT> in <TT>SpecieReference</TT>
must refer to the name of a specie defined in the enclosing
<TT>Model</TT>-type structure.  The two fields <TT>stoichiometry</TT> and
<TT>denominator</TT> together set the stoichiometry value for a specie in a
reaction.  Both are integers, and both have default values of
``<TT>1</TT>'' (one).  The use of these separate terms allows a
simulator to employ rational arithmetic computations on the stoichiometry
matrix, potentially reducing round-off errors and other problems during
computations.  Such computations are particularly important when working
with large matrices and calculating such things as elementary modes.

<P>
The following is a simple example of a specie reference in a list of
reactants within a reaction named ``J1'':
<BR>
<IMG
 WIDTH="480" HEIGHT="189" ALIGN="BOTTOM" BORDER="0"
 SRC="img50.gif"
 ALT="\begin{example}
&lt;model&gt;
...
&lt;listOfReactions&gt;
&lt;reaction name=''J1''&gt;
&lt;listOf...
...eactants&gt;
...
&lt;/reaction&gt;
...
&lt;/listOfReactions&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>

<H3><A NAME="SECTION00047200000000000000"></A>
<A NAME="subsec:kinetic-law"></A><BR>
4.7.2 <TT>KineticLaw</TT>
</H3>

<P>
A <TT>kineticLaw</TT> structure describes the rate of the enclosing
reaction.  The use of a <TT>KineticLaw</TT> structure in a <TT>Reaction</TT>
component is optional; however, in general there is no useful default that
can be substituted in place of a missing kinetic law definition in a
reaction.

<P>
The field <TT>formula</TT>, of type <TT>string</TT>, expresses the rate in
<!-- MATH
 $substance/time$
 -->
<IMG
 WIDTH="113" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img51.gif"
 ALT="$ substance/time$"> units.  (Section&nbsp;<A HREF="sbml-level-1.html#sec:formulas">3.5</A> discusses formulas.)
The optional fields <TT>substanceUnits</TT> and <TT>timeUnits</TT>
determine the units of substance and time.  If not set, the units are taken
from the defaults defined by the built-in ``<TT>substance</TT>'' and
``<TT>time</TT>'' of Table&nbsp;<A HREF="sbml-level-1.html#tab:builtin">5</A>.

<P>
A <TT>KineticLaw</TT> type structure can contain zero or more
<TT>Parameter</TT> structures (Section&nbsp;<A HREF="sbml-level-1.html#sec:parameters">4.5</A>) that define
symbols that can be used in the <TT>formula</TT> string.  As discussed in
Section&nbsp;<A HREF="sbml-level-1.html#sec:namespaces">3.4</A>, reactions introduce local namespaces for
parameter names.  Within a <TT>KineticLaw</TT> structure inside a reaction
definition, a parameter whose name is identical to a global parameter
defined in the enclosing <TT>Model</TT>-type structure takes precedence over
that global parameter.

<P>
The following is an example of a <TT>Reaction</TT> structure that defines
the reaction <!-- MATH
 $J_1: \ X_0 \longrightarrow S_1; \ k_1 X_0$
 -->
<IMG
 WIDTH="159" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img52.gif"
 ALT="$ J_1: \ X_0 \longrightarrow S_1; \ k_1 X_0$">.  It demonstrates
the use of specie references and the <TT>KineticLaw</TT> structure:
<BR>
<IMG
 WIDTH="480" HEIGHT="278" ALIGN="BOTTOM" BORDER="0"
 SRC="img53.gif"
 ALT="\begin{example}
&lt;model&gt;
\ ...
&lt;listOfReactions&gt;
&lt;reaction name=''J1''&gt;
&lt;listO...
...ers&gt;
&lt;/kineticLaw&gt;
&lt;/reaction&gt;
&lt;/listOfReactions&gt;
...
&lt;/model&gt;
\end{example}">
<BR>

<P>

<H1><A NAME="SECTION00050000000000000000"></A>
<A NAME="sec:xml-rep"></A><BR>
5 Examples of Full Models Encoded in XML Using SBML
</H1>

<P>
In this section, we present several examples of complete models encoded in
XML using SBML Level&nbsp;1.  Our approach to translating the UML-based
structure definitions presented in the previous sections is described
elsewhere&nbsp;(<A
 HREF="sbml-level-1.html#hucka:2000b">Hucka, 2000</A>).  Appendix&nbsp;<A HREF="sbml-level-1.html#apdx:schemas">B</A> gives the full
listing of an XML Schema corresponding to SBML Level&nbsp;1.

<P>

<H2><A NAME="SECTION00051000000000000000">
5.1 A Simple Example Application of SBML</A>
</H2>

<P>
Consider the following hypothetical branched system:
<P></P>
<DIV ALIGN="CENTER"><!-- MATH
 \begin{equation*}
\begin{array}{@{}ccc@{}}
    X_0 & \underrightarrow{k_1 X_0} & S_1 \\\\[-4pt]
    S_1 & \underrightarrow{k_2 S_1} & X_1 \\\\[-4pt]
    S_1 & \underrightarrow{k_3 S_1} & X_2
  \end{array}
\end{equation*}
 -->
<TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="107" HEIGHT="113" BORDER="0"
 SRC="img11.gif"
 ALT="\begin{displaymath}\begin{array}{@{}ccc@{}} X_0 &amp; \underrightarrow{k_1 X_0} &amp; S_...
...\  \\  [-4pt] S_1 &amp; \underrightarrow{k_3 S_1} &amp; X_2 \end{array}\end{displaymath}"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P>

<P>
The following is the main portion of an XML document that encodes the model
shown above:

<P>
<BR>
<IMG
 WIDTH="569" HEIGHT="992" ALIGN="BOTTOM" BORDER="0"
 SRC="img54.gif"
 ALT="\begin{example}
&lt;sbml level=''1'' version=''1''&gt;
&lt;model name=''Branch''&gt;
&lt;note...
...
&lt;/kineticLaw&gt;
&lt;/reaction&gt;
&lt;/listOfReactions&gt;
&lt;/model&gt;
&lt;/sbml&gt;
\end{example}">
<BR>

<P>
The XML encoding shown above is quite straightforward. The outermost
container is a tag, <TT>&lt;smbl&gt;</TT>, that identifies the contents as being
<U>S</U>ystems <U>B</U>iology <U>M</U>arkup
<U>L</U>anguage.  The attributes <TT>level</TT> and <TT>version</TT>
indicate that the content is formatted according to version&nbsp;1 of the
Level&nbsp;1 definition of SBML.  The <TT>version</TT> attribute is present in
case SBML Level&nbsp;1 must be revised in the future to correct errors.

<P>
The next-inner container is a single <TT>&lt;model&gt;</TT> element that serves as
the highest-level object in the model.  The model has a name, ``Branch''.
The model contains one compartment, four species, and three reactions.  The
elements in the <TT>&lt;listOfReactants&gt;</TT> and <TT>&lt;listOfProducts&gt;</TT> in
each reaction refer to the names of elements listed in the
<TT>&lt;listOfSpecies&gt;</TT>.  The correspondences between the various elements
is explicitly stated by the <TT>&lt;specieReference&gt;</TT> elements.

<P>
The model includes a <TT>&lt;notes&gt;</TT> annotation that summarizes the model
in text form, with formatting based on XHTML.  This may be useful for a
software package that is able to read such annotations and, for example,
render them in HTML in a graphical user interface.

<P>

<H2><A NAME="SECTION00052000000000000000"></A>
<A NAME="apdx:units-eg"></A><BR>
5.2 Simple Use of Units Feature in a Model
</H2>

<P>
The following model uses the units features of SBML Level&nbsp;1.  In this
model, the default value of <TT>substance</TT> is changed in the list of
unit definitions to be mole units with a scale factor of <IMG
 WIDTH="24" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img25.gif"
 ALT="$ -3$">, or
millimoles.  This sets the default substance units in the model; components
can override this scale locally.  The <TT>volume</TT> and <TT>time</TT>
built-ins are left to their defaults, ensuring that volume is in liters and
time is in seconds.  The result is that, in this model, kinetic law
formulas define rates in millimoles per second and the specie symbols in
them represent concentration values in millimoles per liter.  All the
<TT>specie</TT> elements set the initial amount of every given specie to <IMG
 WIDTH="12" HEIGHT="13" ALIGN="BOTTOM" BORDER="0"
 SRC="img55.gif"
 ALT="$ 1$">
millimole.  The parameters <TT>Vm</TT> and <TT>Km</TT> are defined to be in
millimoles per liter per second, and milliMolar, respectively.

<P>
<BR>
<IMG
 WIDTH="510" HEIGHT="888" ALIGN="BOTTOM" BORDER="0"
 SRC="img56.gif"
 ALT="\begin{example}
&lt;sbml level=''1'' version=''1''&gt;
\ &lt;model&gt;
\ &lt;listOfUnitDefiniti...
...)/(km + s1)''/&gt;
&lt;/reaction&gt;
&lt;/listOfReactions&gt;
&lt;/model&gt;
&lt;/sbml&gt;
\end{example}">
<BR>

<P>

<H2><A NAME="SECTION00053000000000000000"></A>
<A NAME="subsection:ruleseg"></A><BR>
5.3 A Simple Example Application Using Rules
</H2>

<P>
The following model represents the pathway <!-- MATH
 $X_0 \rightarrow S_1 \rightarrow
S_2 \rightarrow X_1$
 -->
<IMG
 WIDTH="153" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img57.gif"
 ALT="$ X_0 \rightarrow S_1 \rightarrow
S_2 \rightarrow X_1$">, where <!-- MATH
 $S_1 \rightarrow S_2$
 -->
<IMG
 WIDTH="63" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img58.gif"
 ALT="$ S_1 \rightarrow S_2$"> is a fast reaction.  The
reaction <!-- MATH
 $S_1 \rightarrow S_2$
 -->
<IMG
 WIDTH="63" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img58.gif"
 ALT="$ S_1 \rightarrow S_2$"> is not modeled explicitly; instead, the
effect of the reaction is encapsulated in rules.

<P>
<BR>
<IMG
 WIDTH="555" HEIGHT="665" ALIGN="BOTTOM" BORDER="0"
 SRC="img59.gif"
 ALT="\begin{example}
&lt;sbml level=''1'' version=''1''&gt;
\ &lt;model&gt;
&lt;listOfCompartments&gt;...
...a=''k4 * s2''/&gt;
&lt;/reaction&gt;
&lt;/listOfReactions&gt;
&lt;/model&gt;
&lt;/sbml&gt;
\end{example}">
<BR>

<P>

<H1><A NAME="SECTION00060000000000000000"></A>
<A NAME="sec:discussion"></A><BR>
6 Discussion
</H1>

<P>
The volume of data now emerging from molecular biotechnology
leave little doubt that extensive computer-based modeling, simulation and
analysis will be critical to understanding and interpreting the
data&nbsp;(<A NAME="tex2html145"
  HREF="#gilman:2000">Gilman, 2000</A>; <A NAME="tex2html146"
  HREF="#abbott:1999">Abbott, 1999</A>; <A NAME="tex2html147"
  HREF="#smaglik:2000">Smaglik, 2000a</A>; <A NAME="tex2html148"
  HREF="#popel:1998">Popel and Winslow, 1998</A>).  This
has lead to an explosion in the development of computer tools by many
research groups across the world.  The explosive rate of progress is
exciting, but the rapid growth of the field is accompanied by problems and
pressing needs.

<P>
One problem is that simulation models and results often cannot be directly
compared, shared or re-used, because the tools developed by different
groups often are not compatible with each other.  As the field of systems
biology matures, researchers increasingly need to communicate their results
as computational models rather than box-and-arrow diagrams.  They also need
to reuse published and curated models as library components in order to
succeed with large-scale efforts&nbsp;(e.g., the Alliance for Cellular
Signaling; <A NAME="tex2html149"
  HREF="#gilman:2000">Gilman, 2000</A>; <A NAME="tex2html150"
  HREF="#smaglik:2000">Smaglik, 2000a</A>).  These needs require that models
implemented in one software package be portable to other software packages,
to maximize public understanding and to allow building up libraries of
curated computational models.

<P>
We offer SBML to the systems biology community as a suggested format for
exchanging models between simulation/analysis tools.  SBML is an open model
representation language oriented specifically towards representing
biochemical network models.  SBML Level&nbsp;1 provides the most basic
facilities that are necessary for expressing these kinds of models in terms
of compartments, species, reactions, parameters, rules and units.

<P>
Our vision for SBML is to create an open standard that will enable
simulation software to exchange models.  SBML is not static; we continue to
develop and experiment with it, and we interact with other groups who seek
to develop similar markup languages.  We plan on continuing to evolve SBML
with the help of the systems biology community to make SBML increasingly
more powerful, flexible and useful.

<P>

<H2><A NAME="SECTION00061000000000000000"></A>
<A NAME="sec:level-2"></A><BR>
6.1 Future Enhancements to SBML: Level 2 and Beyond
</H2>

<P>
As mentioned above, SBML Level 1 is intended to provide the most basic
foundations for modeling biochemical networks.  A number of significant
capabilities are lacking from Level&nbsp;1; these will be introduced in
higher-level definitions of SBML.  The following summarizes additional
features that will likely be included in SBML Level&nbsp;2:

<UL>
<LI><I>Arrays</I>.  This will enable the creation of arrays of components
  (species, reactions, compartments and submodels).

<P>
</LI>
<LI><I>Connections</I>.  This will be a mechanism for describing the
  connections between items in an array.  For example, it should be
  possible to create a 2-D array of compartments and then a 3-D array of
  reactions which transport species between the compartments, where the
  third dimension is the connections between the compartments.  Two
  possible ways of describing a connection scheme are: (1) sparse/explicit,
  simply listing the relative co-coordinates of connected objects for
  patterns of points; (2) algebraic, where a conditional equation describes
  whether two objects are connected.

<P>
</LI>
<LI><I>Database Interoperability</I>.  In order to store models in a
  database, it will be necessary to add additional header information that
  provides information about authors, version numbers, revision dates, etc.

<P>
</LI>
<LI><I>Geometry</I>.  We will develop a scheme for representing the 3-D
  structure of compartments.

<P>
</LI>
<LI><I>Submodels</I>.  This will enable a large model to be built up out
  of instances of other models.  It will also allow the reuse of model
  components and the creation of several instances of the same model.

<P>
</LI>
<LI><I>Component Identification</I>.  This will enable components to be
  described using some stable universal identification scheme.

<P>
</LI>
<LI><I>References</I>.  This will enable literature/authors to be cited
  for any component.

<P>
</LI>
<LI><I>Diagrams</I>.  This feature will allow components to be annotated
  with data to enable the display of the model in a diagram.  It will also
  enable multistate representations.

<P>
</LI>
</UL>

<P>

<H2><A NAME="SECTION00062000000000000000"></A>
<A NAME="sec:other-efforts"></A><BR>
6.2 Relationships to Other Efforts
</H2>

<P>
There are a number of ongoing efforts with similar goals as those of SBML.
Many of them are oriented more specifically toward describing protein
sequences, genes and related entities for database storage and search.
These are generally not intended to be computational models, in the sense
that they do not describe entities and behavioral rules in such a way that
a simulation package could ``run'' the models.

<P>
The effort perhaps closest in spirit to SBML is
CellML&nbsp;(<A
 HREF="sbml-level-1.html#physiome:2001">Physiome Sciences, 2001</A>).  CellML is an XML-based markup language
designed for storing and exchanging computer-based biological models.  It
includes facilities for representing model structure, mathematics and
additional information for database storage and search.  Models are
described in terms of networks of connections between discrete components,
where a component is a functional unit that may correspond to a physical
compartment or simply a convenient modeling abstraction.  Components
contain variables and connections contain mappings between the variables of
connected components.  CellML provides facilities for grouping components
and specifying the kinds of relationships that may exist between
components.  It also uses MathML&nbsp;(<A
 HREF="sbml-level-1.html#w3c:2000b">W3C, 2000b</A>) for expressing mathematical
relationships between components and provides the ability to use ECMAScript
(formerly known as JavaScript) to define functions.

<P>
The constructs in CellML tend to be at a more abstract and general level
than those in SBML Level&nbsp;1, and describes the structure and underlying
mathematics of cellular models in a very general way.  By contrast, SBML is
closer to the internal object model used in model analysis software.
Because SBML Level&nbsp;1 is being developed in the context of interacting with
a number of existing simulation packages, it is a more concrete language
than CellML and may be better suited to its purpose of enabling
interoperability with existing simulation tools.  However, CellML offers
viable alternative ideas and the developers of SBML and CellML are actively
engaged in ensuring that the two representations can be translated between
each other.

<P>

<H2><A NAME="SECTION00063000000000000000"></A>
<A NAME="sec:tracking-xml"></A><BR>
6.3 Tracking the XML Schema Standard
</H2>

<P>
One of the problems in attempting to define an XML Schema for SBML is that,
at the time of this writing, the XML Schema
specification&nbsp;(<A NAME="tex2html151"
  HREF="#biron:2000">Biron and Malhotra, 2000</A>; <A NAME="tex2html152"
  HREF="#thompson:2000">Thompson et&nbsp;al., 2000</A>) has not actually been
finalized.  This has been another motivation for defining SBML in terms of
abstract data structures in a UML-based notation rather than directly as an
XML Schema.

<P>
The moving-target status of the XML Schema standard definition requires
that we plan to update the Schema corresponding to SBML.  The following
is our planned approach for handling changes in the Schema standard:

<OL>
<LI>The definition of SBML Level&nbsp;1 in this document is independent of XML
  Schema.  Therefore, the definition of SBML Level&nbsp;1 expressed here can
  remain the same regardless of what happens to the exact form of XML
  Schema.  Among other benefits, this allows developers to leave their
  programs' internal data structures unchanged in the face of possible
  revisions in the Schema standard.

<P>
</LI>
<LI>In Appendix&nbsp;<A HREF="sbml-level-1.html#apdx:schemas">B</A>, we provide an XML Schema
  corresponding to SBML Level&nbsp;1 that has been created using the current
  definition of XML Schema from the W3C
  Organization&nbsp;(<A NAME="tex2html153"
  HREF="#biron:2000">Biron and Malhotra, 2000</A>; <A NAME="tex2html154"
  HREF="#thompson:2000">Thompson et&nbsp;al., 2000</A>).

<P>
</LI>
<LI>Whenever the definition of XML Schema is updated by the W3C in the
  future, we will issue a revised version of the XML Schema for SBML
  Level&nbsp;1 that conform to the updated standard.  We will leave the previous
  versions still available for reference.  The updated XML Schemas for SBML
  Level&nbsp;1 will be identical to the previous versions except where changes
  in XML Schema force a change in the definition of the Schema for SBML
  Level&nbsp;1.

<P>
</LI>
</OL>

<P>

<H2><A NAME="SECTION00064000000000000000"></A>
<A NAME="sec:availability"></A><BR>
6.4 Availability
</H2>

<P>
The SBML Level 1 definition, the XML Schema corresponding to SBML Level&nbsp;1,
and other related documents are openly available from the Caltech ERATO web
site, <TT><A NAME="tex2html17"
  HREF="http://www.cds.caltech.edu/erato/">http://www.cds.caltech.edu/erato/</A></TT>.

<P>

<H1><A NAME="SECTION00070000000000000000"></A>
<A NAME="sec:acknowledgements"></A><BR>
7 Acknowledgments
</H1>

<P>
SBML was first conceived at the JST/ERATO-sponsored <I>First Workshop on
  Software Platforms for Molecular Biology</I>, held in April, 2000, at the
California Institute of Technology in Pasadena, California, USA.  The
participants collectively decided to begin developing a common XML-based
declarative language for representing models.  A draft version of the
Systems Biology Markup Language was developed by the Caltech ERATO team and
delivered to all collaborators in August, 2000.  This draft version
underwent extensive discussion over mailing lists and then again during the
<I>Second Workshop on Software Platforms for Molecular Biology</I> held in
Tokyo, Japan, November 2000.  A revised version of SBML was issued by the
Caltech ERATO team in December, 2000, and after further discussions over
mailing lists and in meetings, we produced the final version of SBML
Level&nbsp;1 described in this document.

<P>
SBML Level 1 was developed with the help of many people, especially the
authors of BioSpice, DBSolve, E-Cell, Gepasi, StochSim, and Virtual Cell,
and members of the <TT>sysbio</TT> mailing list.  We are particularly
grateful to the following people for discussions and knowledge: Dennis
Bray, Athel Cornish-Bowden, David Fell, Carl Firth, Warren Hedley, Martin
Ginkel, Igor Goryanin, Jay Kaserger, Andreas Kremling, Nicolas Le
Nov&#232;re, Les Loew, Daniel Lucio, Pedro Mendes, Eric Mjolsness, James
Schaff, Bruce Shapiro, Tom Shimizu, Hugh Spence, Joerg Stelling, Kouichi
Takahashi, Masaru Tomita, and John Wagner.

<P>
We are indebted to Daniel Lucio of the Virtual Cell group for generating
the XML Schema of SBML Level&nbsp;1 presented in Appendix&nbsp;<A HREF="sbml-level-1.html#apdx:schemas">B</A>.

<P>

<H1><A NAME="SECTION00080000000000000000">
8 Appendix</A>
</H1>

<H1><A NAME="SECTION00090000000000000000"></A>
<A NAME="apdx:notation"></A><BR>
A. Summary of Notation
</H1>

<P>
The definitive explanation for the notation used in this document can be
found in the companion notation document&nbsp;(Hucka, 2000).  Here we briefly
summarize some of the main components of the notations used in describing
SBML.

<P>
Within the definitions of the various object classes introduced in this
document, the following types of expressions are used many times:

<P>
<BR>
<IMG
 WIDTH="345" HEIGHT="57" ALIGN="BOTTOM" BORDER="0"
 SRC="img60.gif"
 ALT="\begin{example}
field1 : float
field2 : integer[0..*]
field3 : (XHTML)
field4 : float \{use = ''default'' value = ''0.0''\}
\end{example}">
<BR>

<P>
The symbols <TT>field1</TT>, <TT>field2</TT>, etc., represents fields in a
data structure.  The colon immediately after the name separates the name of
the attribute from the type of data that it stores.

<P>
More complex specifications use square brackets (<TT>[]</TT>) just after a
type name.  This is used to indicate that the field contains a list of
elements.  Specifically, the notation <TT>[0..*]</TT> signifies a list
containing zero or more elements; the notation <TT>[1..*]</TT> signifies a
list containing at least one element; and so on.  The approach used here to
translate from a list form into XML is, first, create a subelement named
<TT>listOfs</TT>, where the blank indicates the
capitalized name of the field, and then put a list of elements named after
the field as the content of the <TT>listOfs</TT>
element.

<P>
A field whose type is shown in parentheses is implemented as as an XML
subelement rather than an XML attribute.  The parentheses indicate that the
type refers to the type of the subelement value.

<P>
Expressions in curly braces (<TT>{}</TT>) shown after an attribute type
indicate additional constraints placed on the field.  We express constraints
using XML Schema language.  In the examples above, the expression
<TT>{use="default" value="0.0"}</TT> indicates that the field <TT>field4</TT>
is optional and that it has a default value of <IMG
 WIDTH="24" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img61.gif"
 ALT="$ 0.0$">.

<P>

<H1><A NAME="SECTION000100000000000000000"></A>
<A NAME="apdx:schemas"></A><BR>
B. XML Schema for SBML
</H1>

<P>
The following is an XML Schema definition for the Systems Biology Markup
Language.  Example applications of this XML Schema are presented in
Section&nbsp;<A HREF="sbml-level-1.html#sec:xml-rep">5</A>.

<P>
<FONT SIZE="-1"><BR></FONT><PRE>
&lt;?xml version="1.0" encoding="UTF-8"?&gt;
&lt;xsd:schema xmlns:xsd="http://www.w3.org/2000/10/XMLSchema"&gt;
  &lt;xsd:annotation&gt;
    &lt;xsd:documentation&gt; 
      File name : sbml.xsd
      Author : Daniel Lucio, James Schaff, Andrew Finney, Michael Hucka
      Description : XML Schema for the Systems Biology Markup Language
      Organization: Caltech ERATO Kitano
      Version : 1
      Modified : 2001-03-02 22:29 PST
    &lt;/xsd:documentation&gt;
  &lt;/xsd:annotation&gt;
  &lt;!-- SName --&gt;
  &lt;xsd:simpleType name="SName"&gt;
    &lt;xsd:annotation&gt;
      &lt;xsd:documentation&gt;
         SName type used for expressing
         names of components in a model
      &lt;/xsd:documentation&gt;
    &lt;/xsd:annotation&gt;
    &lt;xsd:restriction base="xsd:string"&gt;
      &lt;xsd:pattern value="(_|[a-z]|[A-Z])(_|[a-z]|[A-Z]|[0-9])*"/&gt;
    &lt;/xsd:restriction&gt;
  &lt;/xsd:simpleType&gt;
  &lt;!-- SBase --&gt;
  &lt;xsd:complexType name="SBase" abstract="true"&gt;
    &lt;xsd:annotation&gt;
      &lt;xsd:documentation&gt;
        Designed to allow a modeler or a pacckage to attach
        information to each component.
      &lt;/xsd:documentation&gt;
    &lt;/xsd:annotation&gt;
    &lt;xsd:sequence&gt;
      &lt;xsd:element name="notes" minOccurs="0"&gt;
        &lt;xsd:complexType&gt;
          &lt;xsd:sequence&gt;
            &lt;xsd:any namespace="http://www.w3.org/1999/xhtml"
                     processContents="skip" maxOccurs="unbounded"/&gt;
          &lt;/xsd:sequence&gt;
        &lt;/xsd:complexType&gt;
      &lt;/xsd:element&gt;
      &lt;xsd:element name="annotations" minOccurs="0"&gt;
        &lt;xsd:complexType&gt;
          &lt;xsd:sequence&gt;
            &lt;xsd:any processContents="skip" maxOccurs="unbounded"/&gt;
          &lt;/xsd:sequence&gt;
        &lt;/xsd:complexType&gt;
      &lt;/xsd:element&gt;
    &lt;/xsd:sequence&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- LisOfParameter  --&gt;
  &lt;xsd:element name="listOfParameters"&gt;
    &lt;xsd:complexType&gt;
      &lt;xsd:sequence&gt;
        &lt;xsd:element name="parameter" type="Parameter"
                     maxOccurs="unbounded"/&gt;
      &lt;/xsd:sequence&gt;
    &lt;/xsd:complexType&gt;
  &lt;/xsd:element&gt;
  &lt;!-- Model --&gt;
  &lt;xsd:complexType name="Model"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:sequence&gt;
          &lt;xsd:element name="listOfUnitDefinitions" minOccurs="0"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element name="unitDefinition" type="UnitDefinition"
                             maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
          &lt;xsd:element name="listOfCompartments"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element name="compartment" type="Compartment" maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
          &lt;xsd:element name="listOfSpecies"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element name="specie" type="Specie" maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
          &lt;xsd:element ref="listOfParameters" minOccurs="0"/&gt;
          &lt;xsd:element name="listOfRules" minOccurs="0"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:choice maxOccurs="unbounded"&gt;
                &lt;xsd:element name="algebraicRule" type="AlgebraicRule" minOccurs="0"/&gt;
                &lt;xsd:element name="compartmentVolumeRule" type="CompartmentVolumeRule"
                             minOccurs="0"/&gt;
                &lt;xsd:element name="specieConcentrationRule" type="SpecieConcentrationRule" 
                             minOccurs="0"/&gt;
                &lt;xsd:element name="parameterRule" type="ParameterRule" minOccurs="0"/&gt;
              &lt;/xsd:choice&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
          &lt;xsd:element name="listOfReactions"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element name="reaction" type="Reaction" maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
        &lt;/xsd:sequence&gt;
        &lt;xsd:attribute name="name" type="SName" use="optional"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- UnitDefinition --&gt;
  &lt;xsd:complexType name="UnitDefinition"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:sequence&gt;
          &lt;xsd:element name="listOfUnits" minOccurs="0"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element name="unit" type="Unit" maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
        &lt;/xsd:sequence&gt;
        &lt;xsd:attribute name="name" type="SName" use="required"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- Compartment --&gt;
  &lt;xsd:complexType name="Compartment"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:attribute name="name" type="SName" use="required"/&gt;
        &lt;xsd:attribute name="volume" type="xsd:double" use="default" value="1"/&gt;
        &lt;xsd:attribute name="units" type="SName" use="optional"/&gt;
        &lt;xsd:attribute name="outside" type="SName" use="optional"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- Specie --&gt;
  &lt;xsd:complexType name="Specie"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:attribute name="name" type="SName" use="required"/&gt;
        &lt;xsd:attribute name="compartment" type="SName" use="required"/&gt;
        &lt;xsd:attribute name="initialAmount" type="xsd:double" use="required"/&gt;
        &lt;xsd:attribute name="units" type="SName" use="optional"/&gt;
        &lt;xsd:attribute name="boundaryCondition" type="xsd:boolean" 
                       use="default" value="false"/&gt;
        &lt;xsd:attribute name="charge" type="xsd:integer" use="optional"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- Parameter --&gt;
  &lt;xsd:complexType name="Parameter"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:attribute name="name" type="SName" use="required"/&gt;
        &lt;xsd:attribute name="value" type="xsd:double" use="required"/&gt;
        &lt;xsd:attribute name="units" type="SName" use="optional"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- UnitKind --&gt;
  &lt;xsd:simpleType name="UnitKind"&gt;
    &lt;xsd:restriction base="xsd:string"&gt;
      &lt;xsd:enumeration value="ampere"/&gt;
      &lt;xsd:enumeration value="becquerel"/&gt;
      &lt;xsd:enumeration value="candela"/&gt;
      &lt;xsd:enumeration value="celsius"/&gt;
      &lt;xsd:enumeration value="coulomb"/&gt;
      &lt;xsd:enumeration value="dimensionless"/&gt;
      &lt;xsd:enumeration value="farad"/&gt;
      &lt;xsd:enumeration value="gram"/&gt;
      &lt;xsd:enumeration value="gray"/&gt;
      &lt;xsd:enumeration value="henry"/&gt;
      &lt;xsd:enumeration value="hertz"/&gt;
      &lt;xsd:enumeration value="item"/&gt;
      &lt;xsd:enumeration value="joule"/&gt;
      &lt;xsd:enumeration value="katal"/&gt;
      &lt;xsd:enumeration value="kelvin"/&gt;
      &lt;xsd:enumeration value="kilogram"/&gt;
      &lt;xsd:enumeration value="liter"/&gt;
      &lt;xsd:enumeration value="litre"/&gt;
      &lt;xsd:enumeration value="lumen"/&gt;
      &lt;xsd:enumeration value="lux"/&gt;
      &lt;xsd:enumeration value="meter"/&gt;
      &lt;xsd:enumeration value="metre"/&gt;
      &lt;xsd:enumeration value="mole"/&gt;
      &lt;xsd:enumeration value="newton"/&gt;
      &lt;xsd:enumeration value="ohm"/&gt;
      &lt;xsd:enumeration value="pascal"/&gt;
      &lt;xsd:enumeration value="radian"/&gt;
      &lt;xsd:enumeration value="second"/&gt;
      &lt;xsd:enumeration value="siemens"/&gt;
      &lt;xsd:enumeration value="sievert"/&gt;
      &lt;xsd:enumeration value="steradian"/&gt;
      &lt;xsd:enumeration value="tesla"/&gt;
      &lt;xsd:enumeration value="volt"/&gt;
      &lt;xsd:enumeration value="watt"/&gt;
      &lt;xsd:enumeration value="weber"/&gt;
    &lt;/xsd:restriction&gt;
  &lt;/xsd:simpleType&gt;
  &lt;!-- Unit --&gt;
  &lt;xsd:complexType name="Unit"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:attribute name="kind" type="UnitKind" use="required"/&gt;
        &lt;xsd:attribute name="exponent" type="xsd:integer" use="default" value="1"/&gt;
        &lt;xsd:attribute name="scale" type="xsd:integer" use="default" value="1"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- Rule --&gt;
  &lt;xsd:simpleType name="RuleType"&gt;
    &lt;xsd:restriction base="xsd:string"&gt;
      &lt;xsd:enumeration value="scalar"/&gt;
      &lt;xsd:enumeration value="rate"/&gt;
    &lt;/xsd:restriction&gt;
  &lt;/xsd:simpleType&gt;
  &lt;xsd:complexType name="Rule" abstract="true"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:attribute name="formula" type="xsd:string" use="required"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:complexType name="AlgebraicRule"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="Rule"/&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:complexType name="AssignmentRule" abstract="true"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="Rule"&gt;
        &lt;xsd:attribute name="type" type="RuleType" use="default" value="scalar"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:complexType name="CompartmentVolumeRule"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="AssignmentRule"&gt;
        &lt;xsd:attribute name="compartment" type="SName" use="required"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:complexType name="SpecieConcentrationRule"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="AssignmentRule"&gt;
        &lt;xsd:attribute name="specie" type="SName" use="required"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:complexType name="ParameterRule"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="AssignmentRule"&gt;
        &lt;xsd:attribute name="name" type="SName" use="required"/&gt;
        &lt;xsd:attribute name="units" type="SName" use="optional"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- Reaction --&gt;
  &lt;xsd:element name="specieReference"&gt;
    &lt;xsd:complexType&gt;
      &lt;xsd:complexContent&gt;
        &lt;xsd:extension base="SBase"&gt;
          &lt;xsd:attribute name="specie" type="xsd:string" use="required"/&gt;
          &lt;xsd:attribute name="stoichiometry" type="xsd:integer" use="default" value="1"/&gt;
          &lt;xsd:attribute name="denominator" type="xsd:integer" use="default" value="1"/&gt;
        &lt;/xsd:extension&gt;
      &lt;/xsd:complexContent&gt;
    &lt;/xsd:complexType&gt;
  &lt;/xsd:element&gt;
  &lt;xsd:complexType name="Reaction"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:sequence&gt;
          &lt;xsd:element name="listOfReactants"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element ref="specieReference" maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
          &lt;xsd:element name="listOfProducts"&gt;
            &lt;xsd:complexType&gt;
              &lt;xsd:sequence&gt;
                &lt;xsd:element ref="specieReference" maxOccurs="unbounded"/&gt;
              &lt;/xsd:sequence&gt;
            &lt;/xsd:complexType&gt;
          &lt;/xsd:element&gt;
          &lt;xsd:element name="kineticLaw" type="KineticLaw" minOccurs="0"/&gt;
        &lt;/xsd:sequence&gt;
        &lt;xsd:attribute name="name" type="SName" use="required"/&gt;
        &lt;xsd:attribute name="reversible" type="xsd:boolean" use="default" value="true"/&gt;
        &lt;xsd:attribute name="fast" type="xsd:boolean" use="default" value="false"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:complexType name="KineticLaw"&gt;
    &lt;xsd:complexContent&gt;
      &lt;xsd:extension base="SBase"&gt;
        &lt;xsd:sequence&gt;
          &lt;xsd:element ref="listOfParameters" minOccurs="0"/&gt;
        &lt;/xsd:sequence&gt;
        &lt;xsd:attribute name="formula" type="xsd:string" use="required"/&gt;
        &lt;xsd:attribute name="timeUnits" type="SName" use="optional"/&gt;
        &lt;xsd:attribute name="substanceUnits" type="SName" use="optional"/&gt;
      &lt;/xsd:extension&gt;
    &lt;/xsd:complexContent&gt;
  &lt;/xsd:complexType&gt;
  &lt;!-- Top-level elements allowed in an SBML document. --&gt;
  &lt;xsd:complexType name="sbmlDocument"&gt;
    &lt;xsd:sequence&gt;
      &lt;xsd:element name="model" type="Model"/&gt;
    &lt;/xsd:sequence&gt;
    &lt;xsd:attribute name="xmlns"/&gt;
    &lt;xsd:attribute name="level" type="xsd:positiveInteger" use="required"/&gt;
    &lt;xsd:attribute name="version" type="xsd:positiveInteger" use="required"/&gt;
  &lt;/xsd:complexType&gt;
  &lt;xsd:element name="sbml" type="sbmlDocument"/&gt;
  &lt;!-- The end. --&gt;
&lt;/xsd:schema&gt;
</PRE>

<P>

<P>

<H1><A NAME="SECTION000110000000000000000"></A>
<A NAME="apdx:predefined-functions"></A><BR>
C. Predefined Functions in SBML
</H1>

<P>
Table&nbsp;<A HREF="sbml-level-1.html#tab:simplemath">6</A> lists the basic mathematical functions that are
defined in SBML Level&nbsp;1 at this time.

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="769"></A>
<TABLE>
<CAPTION><STRONG>Table 6:</STRONG>
Basic mathematical functions defined in SBML.</CAPTION>
<TR><TD> <DIV ALIGN="CENTER">
<TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>&nbsp;</TD>
<TH ALIGN="LEFT" VALIGN="TOP" WIDTH=113><B>Argument</B></TH>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28><B>Name</B></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><B>Args.</B></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142><B>Formula or Meaning</B></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><B>Constraints</B></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108><B>Result Constraints</B></TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>abs</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>absolute value of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>acos</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>arc cosine of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> in radians</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><!-- MATH
 $-1.0 \leq x \leq 1.0$
 -->
<IMG
 WIDTH="109" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img62.gif"
 ALT="$ -1.0 \leq x \leq 1.0$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108><!-- MATH
 $0 \leq acos(x) \leq \pi$
 -->
<IMG
 WIDTH="116" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img63.gif"
 ALT="$ 0 \leq acos(x) \leq \pi$"></TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>asin</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>arc sine of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> in radians</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><!-- MATH
 $-1.0 \leq x \leq 1.0$
 -->
<IMG
 WIDTH="109" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img62.gif"
 ALT="$ -1.0 \leq x \leq 1.0$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108><!-- MATH
 $-\pi/2 \leq asin(x) \leq \pi/2$
 -->
<IMG
 WIDTH="162" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img64.gif"
 ALT="$ -\pi/2 \leq asin(x) \leq \pi/2$"></TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>atan</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>arc tangent of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> in radians</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108><!-- MATH
 $-\pi/2 \leq atan(x) \leq \pi/2$
 -->
<IMG
 WIDTH="164" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img65.gif"
 ALT="$ -\pi/2 \leq atan(x) \leq \pi/2$"></TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>ceil</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>smallest number not less than <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> whose value is an exact integer</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>cos</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>cosine of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>exp</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142><IMG
 WIDTH="20" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img66.gif"
 ALT="$ e^x$">, where <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img67.gif"
 ALT="$ e$"> is the base of the natural logarithm</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>floor</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>the largest number not greater than <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"> whose value is an exact integer</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>log</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>natural logarithm of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><IMG
 WIDTH="42" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img68.gif"
 ALT="$ x &gt; 0$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>log10</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>base 10 logarithm of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><IMG
 WIDTH="42" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img68.gif"
 ALT="$ x &gt; 0$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>pow</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="29" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img69.gif"
 ALT="$ x, y$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142><IMG
 WIDTH="21" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img70.gif"
 ALT="$ x^y$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>sqr</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142><IMG
 WIDTH="20" HEIGHT="17" ALIGN="BOTTOM" BORDER="0"
 SRC="img71.gif"
 ALT="$ x^2$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>sqrt</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142><IMG
 WIDTH="26" HEIGHT="33" ALIGN="MIDDLE" BORDER="0"
 SRC="img72.gif"
 ALT="$ \sqrt{x}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><IMG
 WIDTH="42" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img73.gif"
 ALT="$ x \geq 0$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108><!-- MATH
 $sqrt(x) \geq 0$
 -->
<IMG
 WIDTH="83" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img74.gif"
 ALT="$ sqrt(x) \geq 0$"></TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>sin</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>sine of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>tan</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23><IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>tangent of <IMG
 WIDTH="13" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img37.gif"
 ALT="$ x$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113><!-- MATH
 $x \neq n \frac{\pi}{2}$
 -->
<IMG
 WIDTH="56" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img75.gif"
 ALT="$ x \neq n \frac{\pi}{2}$">, for odd integer n</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=28>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=23>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=142>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=113>&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=108>&nbsp;</TD>
<TD ALIGN="LEFT">&nbsp;</TD></TR>
</TABLE>
</DIV>
  
  <A NAME="tab:simplemath"></A></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>
Table&nbsp;<A HREF="sbml-level-1.html#tab:ratelaws">7</A> defines the rate law functions available in
formula expressions in SBML.  These were extracted from the Gepasi help
file (3.21).  <A
 HREF="sbml-level-1.html#segel:1993">Segel (1993)</A> provides more information;
<A
 HREF="sbml-level-1.html#hofmeyr:1997">Hofmeyr and Cornish-Bowden (1997)</A> provide specific details on the reversible Hill
equations.

<P>
<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="913"></A>
<TABLE>
<CAPTION><STRONG>Table 7:</STRONG>
Table of rate law functions in SBML.  In all cases, <IMG
 WIDTH="59" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img7.gif"
 ALT="$ K_m &gt; 0$">, <!-- MATH
 $V_x \geq 0$
 -->
<IMG
 WIDTH="51" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img8.gif"
 ALT="$ V_x \geq 0$">, <!-- MATH
 $S
\   \geq 0$
 -->
<IMG
 WIDTH="44" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img9.gif"
 ALT="$ S \geq 0$"> and <IMG
 WIDTH="46" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img10.gif"
 ALT="$ P \geq 0$">.</CAPTION>
<TR><TD> 
 
<TABLE CELLPADDING=3 BORDER="1">
<TR><TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36><B>Name</B></TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <B>Arguments</B> </TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> <B>Meaning</B> </TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><B>Formula</B></TH>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>massi</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="19" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img76.gif"
 ALT="$ S_i$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img77.gif"
 ALT="$ k$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Irreversible Mass Action Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><!-- MATH
 $v = k \prod_i S_i$
 -->
<IMG
 WIDTH="83" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img78.gif"
 ALT="$ v = k \prod_i S_i$"></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>massr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="19" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img76.gif"
 ALT="$ S_i$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img79.gif"
 ALT="$ P_j$">, <IMG
 WIDTH="20" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img80.gif"
 ALT="$ k_1$">, <IMG
 WIDTH="20" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img81.gif"
 ALT="$ k_2$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Reversible Mass Action Kinetics  </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><!-- MATH
 $v = k_1 \prod_i S_i - k_2 \prod_j P_j$
 -->
<IMG
 WIDTH="168" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img82.gif"
 ALT="$ v = k_1 \prod_i S_i - k_2 \prod_j P_j$"></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>uui</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="26" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img84.gif"
 ALT="$ V_m$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Irreversible Simple Michaelis-Menten  </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="93" HEIGHT="51" ALIGN="MIDDLE" BORDER="0"
 SRC="img86.gif"
 ALT="$\displaystyle v = \frac{V_m S}{K_m + S}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>uur</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Uni-Uni Reversible Simple
Michaelis-Menten </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="162" HEIGHT="79" ALIGN="MIDDLE" BORDER="0"
 SRC="img91.gif"
 ALT="$\displaystyle v = \frac{V_f \frac{\displaystyle S}{\displaystyle K_{ms}} - V_r ...
...style S}{\displaystyle K_{ms}} + \frac{\displaystyle P}{\displaystyle K_{mp}} }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>uuhr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img92.gif"
 ALT="$ K_{m1}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img93.gif"
 ALT="$ K_{m2}$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Uni-Uni
Reversible Simple Michaelis-Menten with Haldane adjustment </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="146" HEIGHT="77" ALIGN="MIDDLE" BORDER="0"
 SRC="img95.gif"
 ALT="$\displaystyle v = \frac{\frac{V_f}{K_{m1}} \left(S - \frac{P}{K_{eq}} \right)}{1 + \frac{S}{K_{m1}} + \frac{P}{K_{m2}}}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>isouur</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="27" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img96.gif"
 ALT="$ K_{ii}$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Iso Uni-Uni </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="252" HEIGHT="77" ALIGN="MIDDLE" BORDER="0"
 SRC="img97.gif"
 ALT="$\displaystyle v = \frac{V_f \left(S - \frac{P}{K_{eq}}\right)}{S \left(1 + \frac{P}{K_{ii}}\right) + K_{ms} \left(1 + \frac{P}{K_{mp}}\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>hilli</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img99.gif"
 ALT="$ S_{0.5}$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img100.gif"
 ALT="$ h$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Hill Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="103" HEIGHT="56" ALIGN="MIDDLE" BORDER="0"
 SRC="img101.gif"
 ALT="$\displaystyle v = \frac{V S^h}{S_{0.5}^h + S^h}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>hillr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img99.gif"
 ALT="$ S_{0.5}$">, <IMG
 WIDTH="32" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img102.gif"
 ALT="$ P_{0.5}$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img100.gif"
 ALT="$ h$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Reversible
Hill Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="304" HEIGHT="85" ALIGN="MIDDLE" BORDER="0"
 SRC="img103.gif"
 ALT="$\displaystyle v = \frac{\left(V_f \frac{S}{S_{0.5}} \right) \left(1 - \frac{P}{...
..._{0.5}}\right)^{h-1}}{1 + \left(\frac{S}{S_{0.5}} + \frac{P}{P_{0.5}}\right)^h}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>hillmr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img77.gif"
 ALT="$ k$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img100.gif"
 ALT="$ h$">, <IMG
 WIDTH="14" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img105.gif"
 ALT="$ \alpha$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Reversible Hill
Kinetics with One Modifier </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="304" HEIGHT="85" ALIGN="MIDDLE" BORDER="0"
 SRC="img106.gif"
 ALT="$\displaystyle v = \frac{\left(V_f \frac{S}{S_{0.5}} \right) \left(1 - \frac{P}{...
...} \right) \left(\frac{S}{S_{0.5}} + \frac{P}{P_{0.5}}\right)^{h-1}} {K_1 + K_2}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
<TR>
<TD COLSPAN=2 ALIGN="LEFT"><BR>where<P><BR></TD></TR>
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="335" HEIGHT="85" ALIGN="MIDDLE" BORDER="0"
 SRC="img107.gif"
 ALT="$\displaystyle K_1 = \left(\frac{S}{S_{0.5}} + \frac{P}{P_{0.5}}\right)^h, \quad...
...left(\frac{M}{M_{0.5}}\right)^h}{1 \ + \alpha \left(\frac{M}{M_{0.5}}\right)^h}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=36>hillmmr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img77.gif"
 ALT="$ k$">, <IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img100.gif"
 ALT="$ h$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img108.gif"
 ALT="$ a$">, <IMG
 WIDTH="11" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img109.gif"
 ALT="$ b$">, <IMG
 WIDTH="21" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img110.gif"
 ALT="$ \alpha_1$">, <IMG
 WIDTH="21" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img111.gif"
 ALT="$ \alpha_2$">, <!-- MATH
 $\alpha_{12}$
 -->
<IMG
 WIDTH="28" HEIGHT="28" ALIGN="MIDDLE" BORDER="0"
 SRC="img112.gif"
 ALT="$ \alpha_{12}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108>
Reversible Hill Kinetics with Two Modifiers </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=218><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="304" HEIGHT="85" ALIGN="MIDDLE" BORDER="0"
 SRC="img106.gif"
 ALT="$\displaystyle v = \frac{\left(V_f \frac{S}{S_{0.5}} \right) \left(1 - \frac{P}{...
...} \right) \left(\frac{S}{S_{0.5}} + \frac{P}{P_{0.5}}\right)^{h-1}} {K_1 + K_2}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
<TR>
<TD COLSPAN=2 ALIGN="LEFT"><BR>where<P><BR></TD></TR>
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="166" HEIGHT="61" ALIGN="MIDDLE" BORDER="0"
 SRC="img113.gif"
 ALT="$\displaystyle K_1 = \left(\frac{S}{S_{0.5}} + \frac{P}{P_{0.5}}\right)^h ,$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="228" HEIGHT="85" ALIGN="MIDDLE" BORDER="0"
 SRC="img114.gif"
 ALT="$\displaystyle K_2 = \frac{1 + \left(\frac{Ma}{Ma_{0.5}}\right)^h + \left( \frac...
...a_{0.5}} \right)^h \left( \ \frac{Mb}{Mb_{0.5}} \right)^h \biggr] \end{split} }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
</TABLE>

<A NAME="tab:ratelaws"></A></TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>

<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="1026"></A>
<TABLE>
<CAPTION><STRONG>Table 7:</STRONG>
Table of rate law functions in SBML (continued).  In all cases, <IMG
 WIDTH="59" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img7.gif"
 ALT="$ K_m &gt; 0$">, <!-- MATH
 $V_x \geq 0$
 -->
<IMG
 WIDTH="51" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img8.gif"
 ALT="$ V_x \geq 0$">, <!-- MATH
 $S
\   \geq 0$
 -->
<IMG
 WIDTH="44" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img9.gif"
 ALT="$ S \geq 0$"> and <IMG
 WIDTH="46" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img10.gif"
 ALT="$ P \geq 0$">.</CAPTION>
<TR><TD> 
 
<TABLE CELLPADDING=3 BORDER="1">
<TR><TH ALIGN="LEFT" VALIGN="TOP" WIDTH=36><B>Name</B></TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <B>Arguments</B> </TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> <B>Meaning</B> </TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><B>Formula</B></TH>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>usii</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Substrate Inhibition Kinetics
(Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="185" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img116.gif"
 ALT="$\displaystyle v = V \frac{S/K_m}{1 + S/K_m + S^2/K_i}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>usir</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Substrate
Inhibition Kinetics (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="249" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img117.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} + V_r P/K_{mp}}{1 + S/K_{ms} + P/K_{mp} + S^2/K_i}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>uai</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img118.gif"
 ALT="$ K_{sa}$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img119.gif"
 ALT="$ K_{sc}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Substrate Activation </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="262" HEIGHT="59" ALIGN="MIDDLE" BORDER="0"
 SRC="img120.gif"
 ALT="$\displaystyle v = \frac{V \left( S/K_{sa} \right)^2}{1 + S/K_{sc} + \left( S/K_{sa}\right)^2 \ + S/K_{sa}}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>ucii</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Competitive Inhibition (Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="163" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img121.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + S/K_m + I/K_i}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>ucir</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Competitive Inhibition
(Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="245" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img122.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + \ S/K_{ms} + P/K_{mp} + I/K_i}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>unii</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Noncompetitive Inhibition
(Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="240" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img124.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + I/K_i + S/K_m \left( 1 + I/K_i\right) }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>unir</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Noncompetitive
Inhibition (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="335" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img125.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + \ I/K_i + \left( S/K_{ms} + P/K_{mp} \right) \left( 1 + I/K_i\right) }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>uuci</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Uncompetitive Inhibition
(Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="186" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img126.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + S/K_m \left( 1 + I/K_i\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>uucr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108>
Uncompetitive Inhibition (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="275" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img127.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + \left( S/K_{ms} + P/K_{mp} \right) \left( 1 + I/K_i\right) }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>umi</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img128.gif"
 ALT="$ K_{is}$">, <IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img129.gif"
 ALT="$ K_{ic}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Mixed Inhibition
Kinetics (Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="252" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img130.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + I/K_{is} + S/K_m \left( 1 + I/K_{ic} \right) }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>umr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="51" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img131.gif"
 ALT="$ S, P, I,$"> <IMG
 WIDTH="49" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img132.gif"
 ALT="$ V_f, V_r, $"> <!-- MATH
 $K_{ms}, K_{mp},$
 -->
<IMG
 WIDTH="79" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img133.gif"
 ALT="$ K_{ms}, K_{mp}, $"> <!-- MATH
 $K_{is}, K_{ic}$
 -->
<IMG
 WIDTH="61" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img134.gif"
 ALT="$ K_{is}, K_{ic} $"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Mixed
Inhibition Kinetics (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="341" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img135.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + I/K_{is} + \left( S/K_{ms} + P/K_{mp} \right) \left( 1 + I/K_{ic} \right) }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>uai</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img137.gif"
 ALT="$ K_a$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Specific Activation Kinetics -
irreversible </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="174" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img138.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + S/K_m + K_a/A_c}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>uar</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img137.gif"
 ALT="$ K_a$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Specific
Activation Kinetics (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="251" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img139.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + S/K_{ms} + P/K_{mp} + K_a/A_c}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=36>ucti</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img137.gif"
 ALT="$ K_a$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=108> Catalytic Activation
(Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=223><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="264" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img140.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + K_a/A_c + S/K_m \left( 1 + K_a/A_c\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
</TABLE>
</TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>

<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="1157"></A>
<TABLE>
<CAPTION><STRONG>Table 7:</STRONG>
Table of rate law functions in SBML (continued).  In all cases, <IMG
 WIDTH="59" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img7.gif"
 ALT="$ K_m &gt; 0$">, <!-- MATH
 $V_x \geq 0$
 -->
<IMG
 WIDTH="51" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img8.gif"
 ALT="$ V_x \geq 0$">, <!-- MATH
 $S
\   \geq 0$
 -->
<IMG
 WIDTH="44" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img9.gif"
 ALT="$ S \geq 0$"> and <IMG
 WIDTH="46" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img10.gif"
 ALT="$ P \geq 0$">.</CAPTION>
<TR><TD> 
 
<TABLE CELLPADDING=3 BORDER="1">
<TR><TH ALIGN="LEFT" VALIGN="TOP" WIDTH=32><B>Name</B></TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <B>Arguments</B> </TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> <B>Meaning</B> </TH>
<TH ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><B>Formula</B></TH>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>uctr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img137.gif"
 ALT="$ K_a$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86>
Catalytic Activation (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="353" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img141.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + K_a/A_c + \left( S/K_{ms} + P/K_{mp}\right) \left( 1 + K_a/A_c\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>umai</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="35" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img142.gif"
 ALT="$ Kas$">, <IMG
 WIDTH="34" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img143.gif"
 ALT="$ Kac$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> Mixed Activation
Kinetics (Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="282" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img144.gif"
 ALT="$\displaystyle v = \frac{V S/K_m}{1 + Kas/A_c + S/K_m \left( 1 + Kac/A_c\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>umar</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img145.gif"
 ALT="$ K_{as}$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img146.gif"
 ALT="$ K_{ac}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86>
Mixed Activation Kinetics (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="370" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img147.gif"
 ALT="$\displaystyle v = \frac{V_f S/K_{ms} - V_r P/K_{mp}}{1 + K_{as}/A_c + \left( S/K_{ms} + \ P/K_{mp}\right) \left( 1 + K_{ac}/A_c\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>uhmi</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$">, <IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img148.gif"
 ALT="$ K_d$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img108.gif"
 ALT="$ a$">, <IMG
 WIDTH="11" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img109.gif"
 ALT="$ b$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> General Hyperbolic
Modifier Kinetics (Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="281" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img149.gif"
 ALT="$\displaystyle v = \frac{V S/K_m \left( 1 + b M / (a K_d)\right) }{1 + M/K_d + S/K_m \left( 1 + M/(a K_d)\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>uhmr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img148.gif"
 ALT="$ K_d$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img108.gif"
 ALT="$ a$">, <IMG
 WIDTH="11" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img109.gif"
 ALT="$ b$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86>
General Hyperbolic Modifier Kinetics (Reversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="371" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img150.gif"
 ALT="$\displaystyle v = \frac{\left( V_f S/K_{ms} - V_r P/K_{mp}\right) \left( 1 + b ...
...) }{1 + M/K_d + \left( S/K_{ms} + P/K_{mp} \right) \left( 1 + M/(a K_d)\right)}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>ualii</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$">, <IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$">, <IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$">, <IMG
 WIDTH="24" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img151.gif"
 ALT="$ K_s$">, <IMG
 WIDTH="27" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img96.gif"
 ALT="$ K_{ii}$">, <IMG
 WIDTH="14" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img152.gif"
 ALT="$ n$">, <IMG
 WIDTH="15" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img153.gif"
 ALT="$ L$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> Allosteric inhibition
(Irreversible) </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="245" HEIGHT="59" ALIGN="MIDDLE" BORDER="0"
 SRC="img154.gif"
 ALT="$\displaystyle v = \frac{V \left( 1 + S/K_s\right)^{n-1}}{L \left( 1 + I/K_{ii}\right)^n + \left( 1 + S/K_s \right)^n}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>ordubr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img157.gif"
 ALT="$ K_{ma}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img158.gif"
 ALT="$ K_{mq}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img159.gif"
 ALT="$ K_{ip}$">,
<IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> Ordered Uni Bi Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="161" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img160.gif"
 ALT="$\displaystyle v = \frac{V_f \left( A - P Q/K_{eq}\right)}{\begin{split}\biggl[ ...
...; V_f/(V_r K_{eq}) \left( K_{mq} P + K_{mp} Q + P Q\right) \biggr] \end{split}}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>ordbur</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$">, <IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img157.gif"
 ALT="$ K_{ma}$">, <IMG
 WIDTH="39" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img162.gif"
 ALT="$ Kmb$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img163.gif"
 ALT="$ K_{ia}$">,
<IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> Ordered Bi Uni Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="160" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img164.gif"
 ALT="$\displaystyle v = \frac{V_f \left( A B - P/K_{eq}\right)}{\begin{split}\biggl[ ...
...{eq}) \left( K_{mp} + P \left( 1 + A/K_{ia}\right) \right) \biggr] \end{split}}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>ordbbr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$">, <IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img157.gif"
 ALT="$ K_{ma}$">, <IMG
 WIDTH="35" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img165.gif"
 ALT="$ K_{mb}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img163.gif"
 ALT="$ K_{ia}$">,
<IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img166.gif"
 ALT="$ K_{ib}$">, <IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img159.gif"
 ALT="$ K_{ip}$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> Ordered Bi Bi Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="375" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img167.gif"
 ALT="$\displaystyle v = \frac{V_f \left( A B - P Q/K_{eq}\right) }{A B \left( 1 + P/K_{ip}\right) + K_{mb} (A + K_{ia}) + K_{ma} B + K_1}$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
<TR>
<TD COLSPAN=2 ALIGN="LEFT"><BR>where<P><BR></TD></TR>
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="329" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img168.gif"
 ALT="$\displaystyle K_1 = V_f / (V_r K_{eq}) \left( K_{mq} P \left( 1 + A/K_{ia}\right) + Q k_2 \right) ,$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="363" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img169.gif"
 ALT="$\displaystyle K_2 = K_{mp} \left( 1 + K_{ma} B /(K_{ia} K_{mb}) + P \left( 1 + B/K_{ib} \right) \right)$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
<TR><TD ALIGN="LEFT" VALIGN="TOP" WIDTH=32>ppbr</TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=55> <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$">, <IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$">, <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$">, <IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$">, <IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$">, <IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$">, <IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img157.gif"
 ALT="$ K_{ma}$">, <IMG
 WIDTH="35" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img165.gif"
 ALT="$ K_{mb}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$">, <IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img158.gif"
 ALT="$ K_{mq}$">,
<IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img163.gif"
 ALT="$ K_{ia}$">, <IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img170.gif"
 ALT="$ K_{iq}$">, <IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"> </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=86> Ping Pong Bi Bi Kinetics </TD>
<TD ALIGN="LEFT" VALIGN="MIDDLE" WIDTH=249><P></P>
<DIV ALIGN="CENTER"><TABLE CELLPADDING="0" WIDTH="100%" ALIGN="CENTER">
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="323" HEIGHT="53" ALIGN="MIDDLE" BORDER="0"
 SRC="img171.gif"
 ALT="$\displaystyle v = \frac{V_f \left( A B - P Q /K_{eq} \right) }{A B + K_{mb} A + K_{ma} B \left( 1 + \ Q/K_{iq}\right)+ K_1 }$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
<TR>
<TD COLSPAN=2 ALIGN="LEFT"><BR>where<P><BR></TD></TR>
<TR VALIGN="MIDDLE">
<TD NOWRAP ALIGN="CENTER"><IMG
 WIDTH="383" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img172.gif"
 ALT="$\displaystyle K_1 = V_f/(V_r K_{eq}) \left( K_{mq} P \left( 1 + A/K_{ia} \right) + Q (K_{mp} + P) \right)$"></TD>
<TD NOWRAP WIDTH="10" ALIGN="RIGHT">
&nbsp;&nbsp;&nbsp;</TD></TR>
</TABLE></DIV>
<BR CLEAR="ALL"><P></P></TD>
</TR>
</TABLE>
</TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>

<BR><P></P>
<DIV ALIGN="CENTER"><A NAME="1296"></A>
<TABLE>
<CAPTION><STRONG>Table 8:</STRONG>
Table of symbols used in Table&nbsp;<A HREF="sbml-level-1.html#tab:ratelaws">7</A>.</CAPTION>
<TR><TD> <TABLE CELLPADDING=3>
<TR><TD ALIGN="LEFT"><B>Symbol</B></TD>
<TH ALIGN="LEFT" VALIGN="TOP" WIDTH=396><B>Meaning</B></TH>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="14" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img105.gif"
 ALT="$ \alpha$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Effect of <IMG
 WIDTH="15" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img83.gif"
 ALT="$ S$"> and <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"> on binding of <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$"> (if <IMG
 WIDTH="50" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img173.gif"
 ALT="$ M&lt;1$">, <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$"> is inhibitor; if <IMG
 WIDTH="50" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img174.gif"
 ALT="$ M&gt;1$">, <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$"> is activator)</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>First substrate in two substrate reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img136.gif"
 ALT="$ A_c$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Activator</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Second substrate in two substrate reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img123.gif"
 ALT="$ I$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Inhibitor</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img175.gif"
 ALT="$ K_1$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Forward Rate Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img176.gif"
 ALT="$ K_2$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Reverse Rate Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img137.gif"
 ALT="$ K_a$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Activation Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img146.gif"
 ALT="$ K_{ac}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Catalytic Activation Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img145.gif"
 ALT="$ K_{as}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Specific Activation Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="25" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img148.gif"
 ALT="$ K_d$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Dissociation constant of the elementary step <!-- MATH
 $E + M = EM$
 -->
<IMG
 WIDTH="104" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img177.gif"
 ALT="$ E + M = EM$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img94.gif"
 ALT="$ K_{eq}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Equilibrium Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="27" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img96.gif"
 ALT="$ K_{ii}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Dissociation constant of the inhibitor from the inactive form of the enzyme</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="23" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img115.gif"
 ALT="$ K_i$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Inhibition constant for the substrate.</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img163.gif"
 ALT="$ K_{ia}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Product inhibition constant of <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$"> acting on the reverse reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img166.gif"
 ALT="$ K_{ib}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Product inhibition constant of <IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$"> acting on the reverse reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img129.gif"
 ALT="$ K_{ic}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Catalytic (noncompetitive) inhibition constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img159.gif"
 ALT="$ K_{ip}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Product inhibition constant of <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"> acting on the forward reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img170.gif"
 ALT="$ K_{iq}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Product inhibition constant of <IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$"> acting on the forward reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="29" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img128.gif"
 ALT="$ K_{is}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Specific (competitive) inhibition constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img85.gif"
 ALT="$ K_m$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Forward Michaelis-Menten Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img157.gif"
 ALT="$ K_{ma}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Concentration of <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$"> such that <IMG
 WIDTH="67" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img178.gif"
 ALT="$ v = V_f/2$"> (Michaelis constant) at zero <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"> and zero <IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="35" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img165.gif"
 ALT="$ K_{mb}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Concentration of <IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$"> such that <IMG
 WIDTH="67" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img178.gif"
 ALT="$ v = V_f/2$"> (Michaelis constant) at saturating <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$"> and zero <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Concentration of <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"> such that <!-- MATH
 $v = -V_r/2$
 -->
<IMG
 WIDTH="78" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img179.gif"
 ALT="$ v = -V_r/2$"> (Michaelis constant) at zero <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$"> and <IMG
 WIDTH="17" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img161.gif"
 ALT="$ B$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img90.gif"
 ALT="$ K_{mp}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Product Michaelis-Menten Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img158.gif"
 ALT="$ K_{mq}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Concentration of <IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$"> such that <!-- MATH
 $v = -V_r/2$
 -->
<IMG
 WIDTH="78" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img179.gif"
 ALT="$ v = -V_r/2$"> (Michaelis constant) at zero <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img155.gif"
 ALT="$ A$"> and saturating <IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="36" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img89.gif"
 ALT="$ K_{ms}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Substrate Michaelis-Menten Constant</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="24" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img151.gif"
 ALT="$ K_s$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Dissociation constant of the substrate from the active form of the enzyme</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="31" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img118.gif"
 ALT="$ K_{sa}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Dissociation constant of substrate-activation site</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="30" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img119.gif"
 ALT="$ K_{sc}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Dissociation constant of substrate-active site</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="15" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img153.gif"
 ALT="$ L$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Equilibrium constant between the active and inactive forms of the enzyme</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Modifier</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img180.gif"
 ALT="$ M_{0.5}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Concentration of <IMG
 WIDTH="21" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img104.gif"
 ALT="$ M$"> that half-saturates its binding site when <IMG
 WIDTH="44" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img181.gif"
 ALT="$ S = 0$">, <IMG
 WIDTH="46" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img182.gif"
 ALT="$ P=0$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="16" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img40.gif"
 ALT="$ P$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>First product in two product reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="38" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img183.gif"
 ALT="$ P_{0.05}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Product concent. s.t. <!-- MATH
 $v = - V_r/2$
 -->
<IMG
 WIDTH="78" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img179.gif"
 ALT="$ v = -V_r/2$"> when <IMG
 WIDTH="84" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img184.gif"
 ALT="$ P = M = 0$"> (<IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$"> is limiting rate of reverse reaction)</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="17" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img156.gif"
 ALT="$ Q$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Second product in two product reaction</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="37" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img185.gif"
 ALT="$ S_{0.05}$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Irreversible rate laws: Substrate concentration such that <IMG
 WIDTH="67" HEIGHT="31" ALIGN="MIDDLE" BORDER="0"
 SRC="img178.gif"
 ALT="$ v = V_f/2$"> when <!-- MATH
 $P = 0, M=0$
 -->
<IMG
 WIDTH="99" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img186.gif"
 ALT="$ P = 0, M=0$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="17" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img98.gif"
 ALT="$ V$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Forward Maximum Velocity</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="22" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img87.gif"
 ALT="$ V_f$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Forward Maximum Velocity</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="26" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img84.gif"
 ALT="$ V_m$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Forward Maximum Velocity</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="21" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img88.gif"
 ALT="$ V_r$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Reverse Maximum Velocity</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="12" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img108.gif"
 ALT="$ a$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Ratio of dissociation constant of elementary step <!-- MATH
 $ES + M = ESM$
 -->
<IMG
 WIDTH="126" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img187.gif"
 ALT="$ ES + M = ESM$"> over that of <!-- MATH
 $E + M = EM$
 -->
<IMG
 WIDTH="104" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img177.gif"
 ALT="$ E + M = EM$"></TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="11" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img109.gif"
 ALT="$ b$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Ratio of rate constant of elementary step <!-- MATH
 $ESM \rightarrow EM + P$
 -->
<IMG
 WIDTH="131" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img188.gif"
 ALT="$ ESM \rightarrow EM + P$"> over that of <!-- MATH
 $ES \rightarrow E + P$
 -->
<IMG
 WIDTH="97" HEIGHT="29" ALIGN="MIDDLE" BORDER="0"
 SRC="img189.gif"
 ALT="$ ES \rightarrow E + P$">.</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="13" HEIGHT="15" ALIGN="BOTTOM" BORDER="0"
 SRC="img100.gif"
 ALT="$ h$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>Hill Coefficient</TD>
</TR>
<TR><TD ALIGN="LEFT"><IMG
 WIDTH="14" HEIGHT="14" ALIGN="BOTTOM" BORDER="0"
 SRC="img152.gif"
 ALT="$ n$"></TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>No. binding sites for substrate &amp; inhibitor (typically the number of monomers in the enzyme)</TD>
</TR>
<TR><TD ALIGN="LEFT">&nbsp;</TD>
<TD ALIGN="LEFT" VALIGN="TOP" WIDTH=396>&nbsp;</TD>
</TR>
</TABLE>
 </TD></TR>
</TABLE>
</DIV><P></P><BR>

<P>

<P>

<P>
<H2><A NAME="SECTIONREF">Bibliography</A>
</H2>

<DL COMPACT>
<DT><A NAME="abbott:1999">
Abbott, A. (1999).
</A>
<DD> Alliance of US labs plans to build map of cell signalling pathways.
<BR> <EM>Nature</EM>, 402:219-200.

<BR>
<P>
<DT><A NAME="arkin:2001">
Arkin, A.&nbsp;P. (2001).
</A>
<DD> <I>Simulac</I> and <I>Deduce</I>.
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html23"
  HREF="http://gobi.lbl.gov/~aparkin/Stuff/Software.html">http://gobi.lbl.gov/~aparkin/Stuff/Software.html</A></TT>.

<BR>
<P>
<DT><A NAME="biron:2000">
Biron, P.&nbsp;V. and Malhotra, A. (2000).
</A>
<DD> XML Schema part 2: Datatypes (W3C candidate recommendation 24
  October 2000).
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html24"
  HREF="http://www.w3.org/TR/xmlschema-2/">http://www.w3.org/TR/xmlschema-2/</A></TT>.

<BR>
<P>
<DT><A NAME="bosak:1999">
Bosak, J. and Bray, T. (1999).
</A>
<DD> XML and the second-generation web.
<BR> <EM>Scientific American</EM>.

<BR>
<P>
<DT><A NAME="bray:2001">
Bray, D., Firth, C., Le Nov&#232;re, N., and Shimizu, T. (2001).
</A>
<DD> <I>StochSim</I>.
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html25"
  HREF="http://www.zoo.cam.ac.uk/comp-cell/StochSim.html">http://www.zoo.cam.ac.uk/comp-cell/StochSim.html</A></TT>.

<BR>
<P>
<DT><A NAME="bray:1999">
Bray, T., D.&nbsp;Hollander, D., and Layman, A. (1999).
</A>
<DD> Namespaces in XML. World Wide Web Consortium
  14-January-1999.
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html26"
  HREF="http://www.w3.org/TR/1999/REC-xml-names-19990114/">http://www.w3.org/TR/1999/REC-xml-names-19990114/</A></TT>.

<BR>
<P>
<DT><A NAME="bray:1998">
Bray, T., Paoli, J., and Sperberg-McQueen, C.&nbsp;M. (1998).
</A>
<DD> Extensible markup language (XML) 1.0, W3C recommendation
  10-February-1998.
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html27"
  HREF="http://www.w3.org/TR/1998/REC-xml-19980210">http://www.w3.org/TR/1998/REC-xml-19980210</A></TT>.

<BR>
<P>
<DT><A NAME="eriksson:1998">
Eriksson, H.-E. and Penker, M. (1998).
</A>
<DD> <EM>UML Toolkit</EM>.
<BR> John Wiley &amp; Sons, New York.

<BR>
<P>
<DT><A NAME="fallside:2000">
Fallside, D.&nbsp;C. (2000).
</A>
<DD> XML Schema part 0: Primer (W3C candidate recommendation 24
  October 2000).
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html28"
  HREF="http://www.w3.org/TR/xmlschema-0/">http://www.w3.org/TR/xmlschema-0/</A></TT>.

<BR>
<P>
<DT><A NAME="gilman:2000">
Gilman, A. (2000).
</A>
<DD> A letter to the signaling community.
<BR> Alliance for Cellular Signaling, The University of Texas Southwestern
  Medical Center. Available via the World Wide Web at
  <TT><A NAME="tex2html29"
  HREF="http://afcs.swmed.edu/afcs/Letter_to_community.htm">http://afcs.swmed.edu/afcs/Letter_to_community.htm</A></TT>.

<BR>
<P>
<DT><A NAME="goryanin:2001">
Goryanin, I. (2001).
</A>
<DD> <I>DBSolve</I>: Software for metabolic, enzymatic and receptor-ligand
  binding simulation.
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html30"
  HREF="http://websites.ntl.com/-igor.goryanin/">http://websites.ntl.com/-igor.goryanin/</A></TT>.

<BR>
<P>
<DT><A NAME="goryanin:1999">
Goryanin, I., Hodgman, T.&nbsp;C., and Selkov, E. (1999).
</A>
<DD> Mathematical simulation and analysis of cellular metabolism and
  regulation.
<BR> <EM>Bioinformatics</EM>, 15(9):749-758.

<BR>
<P>
<DT><A NAME="harbison:1995">
Harbison, S.&nbsp;P. and Steele, G.&nbsp;L. (1995).
</A>
<DD> <EM>C: A Reference Manual</EM>.
<BR> Prentice-Hall.

<BR>
<P>
<DT><A NAME="harold:2001">
Harold, E.&nbsp;R. and Means, E.&nbsp;S. (2001).
</A>
<DD> <EM>XML in a Nutshell</EM>.
<BR> O'Reilly &amp; Associates.

<BR>
<P>
<DT><A NAME="hofmeyr:1997">
Hofmeyr, J.&nbsp;H. and Cornish-Bowden, A. (1997).
</A>
<DD> The reversible Hill equation: How to incorporate cooperative
  enzymes into metabolic models.
<BR> <EM>Computer Applications in the Biosciences</EM>, 13:377-385.

<BR>
<P>
<DT><A NAME="hucka:2000b">
Hucka, M. (2000).
</A>
<DD> SCHUCS: A notation for describing model representations intended
  for XML encoding.
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html31"
  HREF="ftp://ftp.cds.caltech.edu/pub/caltech-erato/notation/">ftp://ftp.cds.caltech.edu/pub/caltech-erato/notation/</A></TT>.

<BR>
<P>
<DT><A NAME="kitano:2001">
Kitano, H. (2001).
</A>
<DD> <EM>Foundations of Systems Biology</EM>.
<BR> MIT Press.

<BR>
<P>
<DT><A NAME="mendes:1997">
Mendes, P. (1997).
</A>
<DD> Biochemistry by numbers: Simulation of biochemical pathways with
  Gepasi&nbsp;3.
<BR> <EM>Trends in Biochemical Sciences</EM>, 22:361-363.

<BR>
<P>
<DT><A NAME="mendes:2001">
Mendes, P. (2001).
</A>
<DD> Gepasi 3.21.
<BR> Available via the World Wide Web at <TT><A NAME="tex2html32"
  HREF="http://www.gepasi.org">http://www.gepasi.org</A></TT>.

<BR>
<P>
<DT><A NAME="morton-firth:1998">
Morton-Firth, C.&nbsp;J. and Bray, D. (1998).
</A>
<DD> Predicting temporal fluctuations in an intracellular signalling
  pathway.
<BR> <EM>Journal of Theoretical Biology</EM>, 192:117-128.

<BR>
<P>
<DT><A NAME="naur:1960">
Naur, P. (1960).
</A>
<DD> Revised report on the algorithmic language ALGOL 60.
<BR> <EM>Communications of the ACM</EM>, 3(5):299-314.

<BR>
<P>
<DT><A NAME="oestereich:1999">
Oestereich, B. (1999).
</A>
<DD> <EM>Developing Software with UML: Object-Oriented Analysis and
  Design in Practice</EM>.
<BR> Addison-Wesley Publishing Company.

<BR>
<P>
<DT><A NAME="physiome:2001">
Physiome Sciences, I. (2001).
</A>
<DD> CellML home page.
<BR> Available via the World Wide Web at <TT><A NAME="tex2html33"
  HREF="http://cellml.org/">http://cellml.org/</A></TT>.

<BR>
<P>
<DT><A NAME="popel:1998">
Popel, A. and Winslow, R.&nbsp;L. (1998).
</A>
<DD> A letter from the directors...
<BR> Center for Computational Medicine &amp; Biology, Johns Hopkins School of
  Medicine, Johns Hopkins University. Available via the World Wide Web at
  <TT><A NAME="tex2html34"
  HREF="http://www.bme.jhu.edu/ccmb/ccmbletter.html">http://www.bme.jhu.edu/ccmb/ccmbletter.html</A></TT>.

<BR>
<P>
<DT><A NAME="sauro:2000">
Sauro, H.&nbsp;M. (2000).
</A>
<DD> Jarnac: A system for interactive metabolic analysis.
<BR> In Hofmeyr, J.-H.&nbsp;S., Rohwer, J.&nbsp;M., and Snoep, J.&nbsp;L., editors, <EM>  Animating the Cellular Map: Proceedings of the 9th International Meeting on
  BioThermoKinetics</EM>. Stellenbosch University Press.
<BR> ISBN 0-7972-0776-7.

<BR>
<P>
<DT><A NAME="sauro:1991">
Sauro, H.&nbsp;M. and Fell, D.&nbsp;A. (1991).
</A>
<DD> SCAMP: A metabolic simulator and control analysis program.
<BR> <EM>Mathl. Comput. Modelling</EM>, 15:15-28.

<BR>
<P>
<DT><A NAME="schaff:2000">
Schaff, J., Slepchenko, B., and Loew, L.&nbsp;M. (2000).
</A>
<DD> Physiological modeling with the Virtual Cell framework.
<BR> In Johnson, M. and Brand, L., editors, <EM>Methods in Enzymology</EM>,
  volume 321, pages 1-23. Academic Press, San Diego.

<BR>
<P>
<DT><A NAME="schaff:2001">
Schaff, J., Slepchenko, B., Morgan, F., Wagner, J., Resasco, D., Shin, D.,
  Choi, Y.&nbsp;S., Loew, L., Carson, J., Cowan, A., Moraru, I., Watras, J.,
  Teraski, M., and Fink, C. (2001).
</A>
<DD> Virtual Cell.
<BR> Available via the World Wide Web at <TT><A NAME="tex2html35"
  HREF="http://www.nrcam.uchc.edu">http://www.nrcam.uchc.edu</A></TT>.

<BR>
<P>
<DT><A NAME="segel:1993">
Segel, I.&nbsp;H. (1993).
</A>
<DD> <EM>Enzyme Kinetics</EM>.
<BR> Wiley Classics Library.

<BR>
<P>
<DT><A NAME="smaglik:2000">
Smaglik, P. (2000a).
</A>
<DD> For my next trick...
<BR> <EM>Nature</EM>, 407:828-829.

<BR>
<P>
<DT><A NAME="smaglik:2000b">
Smaglik, P. (2000b).
</A>
<DD> US grant glues 'virtual cell' together.
<BR> <EM>Nature</EM>, 407(6806):819.

<BR>
<P>
<DT><A NAME="taylor:1995">
Taylor, B.&nbsp;N. (1995).
</A>
<DD> Guide for the use of the international system of units (SI).
<BR> NIST Special Publication 811, 1995 Edition. National Institute of
  Standards and Technology, United States Department of Commerce. Available via
  the World Wide Web at <TT><A NAME="tex2html36"
  HREF="http://physics.nist.gov/Pubs/SP811/contents.html">http://physics.nist.gov/Pubs/SP811/contents.html</A></TT>.

<BR>
<P>
<DT><A NAME="thompson:2000">
Thompson, H.&nbsp;S., Beech, D., Maloney, M., and Mendelsohn, N. (2000).
</A>
<DD> XML Schema part 1: Structures (W3C candidate recommendation 24
  October 2000).
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html37"
  HREF="http://www.w3.org/TR/xmlschema-1/">http://www.w3.org/TR/xmlschema-1/</A></TT>.

<BR>
<P>
<DT><A NAME="tomita:1999">
Tomita, M., Hashimoto, K., Takahashi, K., Shimizu, T., Matsuzaki, Y., Miyoshi,
  F., Saito, K., Tanida, S., Yugi, K., Venter, J.&nbsp;C., and Hutchison, C. (1999).
</A>
<DD> E-Cell: Software environment for whole cell simulation.
<BR> <EM>Bioinformatics</EM>, 15(1):72-84.

<BR>
<P>
<DT><A NAME="tomita:2001">
Tomita, M., Nakayama, Y., Naito, Y., Shimizu, T., Hashimoto, K., Takahashi, K.,
  Matsuzaki, Y., Yugi, K., Miyoshi, F., Saito, Y., Kuroki, A., Ishida, T.,
  Iwata, T., Yoneda, M., Kita, M., Yamada, Y., Wang, E., Seno, S., Okayama, M.,
  Kinoshita, A., Fujita, Y., Matsuo, R., Yanagihara, T., Watari, D., Ishinabe,
  S., and Miyamoto, S. (2001).
</A>
<DD> E-Cell.
<BR> Available via the World Wide Web at <TT><A NAME="tex2html38"
  HREF="http://www.e-cell.org/">http://www.e-cell.org/</A></TT>.

<BR>
<P>
<DT><A NAME="unicode:1996">
Unicode Consortium (1996).
</A>
<DD> <EM>The Unicode Standard, Version 2.0</EM>.
<BR> Addison-Wesley Developers Press, Reading, Massachusetts.

<BR>
<P>
<DT><A NAME="w3c:2000">
W3C (2000a).
</A>
<DD> Naming and addressing: URIs, URLs, ...
<BR> Available via the World Wide Web at
  <TT><A NAME="tex2html39"
  HREF="http://www.w3.org/Addressing/">http://www.w3.org/Addressing/</A></TT>.

<BR>
<P>
<DT><A NAME="w3c:2000b">
W3C (2000b).
</A>
<DD> W3C's math home page.
<BR> Available via the World Wide Web at <TT><A NAME="tex2html40"
  HREF="http://www.w3.org/Math/">http://www.w3.org/Math/</A></TT>.

<BR>
<P>
</DL>

<P>

<H1><A NAME="SECTION000130000000000000000">
About this document ...</A>
</H1>
 <STRONG>Systems Biology Markup Language (SBML) Level 1:
<BR>  Structures and Facilities for Basic Model Definitions</STRONG><P>
This document was generated using the
<A HREF="http://www-dsed.llnl.gov/files/programs/unix/latex2html/manual/"><STRONG>LaTeX</STRONG>2<tt>HTML</tt></A> translator Version 99.2beta8 (1.46)
<P>
Copyright &#169; 1993, 1994, 1995, 1996,
<A HREF="http://cbl.leeds.ac.uk/nikos/personal.html">Nikos Drakos</A>, 
Computer Based Learning Unit, University of Leeds.
<BR>Copyright &#169; 1997, 1998, 1999,
<A HREF="http://www.maths.mq.edu.au/~ross/">Ross Moore</A>, 
Mathematics Department, Macquarie University, Sydney.
<P>
The command line arguments were: <BR>
 <STRONG>latex2html</STRONG> <TT>-white -split +0 -show_section_numbers -image_type gif -no_navigation -local_icons -discard -mkdir -dir html sbml-level-1</TT>
<P>
The translation was initiated by  on 2001-03-14<BR><HR>
<ADDRESS>

2001-03-14
</ADDRESS>
</BODY>
</HTML>
