<!-- MHonArc v2.4.6 -->
<!--X-Subject: [sbml&#45;discuss] SBML models of open biocyc databases now available... -->
<!--X-From-R13: mhpxreNerfrnepu.qspv.uneineq.rqh -->
<!--X-Date: Wed, 22 Oct 2003 20:11:32 &#45;0700 -->
<!--X-Message-Id: 1311.151.199.22.192.1066875290.squirrel@research.dfci.harvard.edu -->
<!--X-Content-Type: text/plain -->
<!--X-Head-End-->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML//EN">
<HTML>
<HEAD>
<TITLE>[sbml-discuss] SBML models of open biocyc databases now available...</TITLE>
<LINK REV="made" HREF="mailto:zucker@research.dfci.harvard.edu">
</HEAD>
<BODY BGCOLOR="white">
<!--X-Body-Begin-->
<!--X-User-Header-->
<!--X-User-Header-End-->
<!--X-TopPNI-->
<HR>
[<A HREF="msg00707.html">Date Prev</A>][Date Next][<A HREF="msg00706.html">Thread Prev</A>][Thread Next][<A HREF="maillist.html#00708">Date Index</A>][<A HREF="threads.html#00708">Thread Index</A>]
<!--X-TopPNI-End-->
<!--X-MsgBody-->
<!--X-Subject-Header-Begin-->
<H1>[sbml-discuss] SBML models of open biocyc databases now available...</H1>
<HR>
<!--X-Subject-Header-End-->
<!--X-Head-of-Message-->
<UL>
<LI><em>To</em>: &lt;<A HREF="mailto:sbml-discuss@caltech.edu">sbml-discuss@caltech.edu</A>&gt;</LI>
<LI><em>Subject</em>: [sbml-discuss] SBML models of open biocyc databases now available...</LI>
<LI><em>From</em>: <A HREF="mailto:zucker@research.dfci.harvard.edu">zucker@research.dfci.harvard.edu</A></LI>
<LI><em>Date</em>: Wed, 22 Oct 2003 22:14:50 -0400 (EDT)</LI>
<LI>Cc: &lt;<A HREF="mailto:ptools-users@ai.sri.com">ptools-users@ai.sri.com</A>&gt;</LI>
<LI>Importance: Normal</LI>
</UL>
<!--X-Head-of-Message-End-->
<!--X-Head-Body-Sep-Begin-->
<HR>
<!--X-Head-Body-Sep-End-->
<!--X-Body-of-Message-->
<PRE>
Hello all,

  I don't normally cross-post to two different mailing lists, but in this
case, I thought it appropriate, since the information in this
announcement not only pertains to both of  these communities, but also
attempts to integrate the efforts of the two.

For those of you in the SBML community, one of the primary goals of Biocyc
is to "support computational studies of the metabolism, such as design of
novel biochemical pathways for biotechnology, studies of the evolution of
metabolic pathways, and simulation of metabolic pathways."

For those of you in the Pathway tools community, the Systems Biology
Markup Language "is designed to enable the exchange of biochemical network
models between different software packages. The aim is to (1)  enable the
exchange of models between software tools with little or no human
intervention, thus allowing the tools to be properly integrated; and (2)
enable these models to be published in electronic form alongside, for
example, peer-reviewed journal articles. The emphasis in SBML is on
supporting quantitative models."

Now that both communities have been introduced, I would like to announce
that SBML models generated from the following open (i.e. available without
restriction) pathway/genome databases are now available:


    * AgroCyc -- Agrobacterium tumefaciens
    * HpyCyc -- Helicobacter pylori
    * MtbCdcCyc -- Mycobacterium tuberculosis CDC1551
    * MtbRvCyc -- Mycobacterium tuberculosis H37Rv
    * PseudoCyc -- Pseudomonas aeruginosa
    * VchoCyc -- Vibrio cholerae

Every reaction is linked back to its full description at Biocyc.org
These models can currently be downloaded from
<A  HREF="http://genome.dfci.harvard.edu/~zucker/BPHYS/biocyc-open/">http://genome.dfci.harvard.edu/~zucker/BPHYS/biocyc-open/</A>
but will eventually move to <A  HREF="http://www.sbml.org/models/">http://www.sbml.org/models/</A>


The second thing I would like to announce is the availability of
Biocyc2SBML, a tool for generating models of metabolism from any
pathway/genome database, including:

    * AgroCyc -- Agrobacterium tumefaciens
    * BsubCyc -- Bacillus subtilis
    * CauloCyc -- Caulobacter crescentus
    * CtraCyc -- Chlamydia trachomatis
    * EcoCyc -- Escherichia coli
    * HinCyc -- Haemophilus influenzae
    * HpyCyc -- Helicobacter pylori
    * MetaCyc -- Metabolic pathways and enzymes from 150 species
    * MtbCdcCyc -- Mycobacterium tuberculosis CDC1551
    * MtbRvCyc -- Mycobacterium tuberculosis H37Rv
    * MpneuCyc -- Mycoplasma pneumoniae
    * PseudoCyc -- Pseudomonas aeruginosa
    * YeastCyc -- Saccharomyces cerevisiae
    * TpalCyc -- Treponema pallidum
    * VchoCyc -- Vibrio cholerae

Biocyc2SBML is also currently available at
<A  HREF="http://genome.dfci.harvard.edu/~zucker/BPHYS/biocyc-open/">http://genome.dfci.harvard.edu/~zucker/BPHYS/biocyc-open/</A>

Furthermore, Biocyc2SBML depends on the Biocyc pathway tools suite
distributed by SRI.
  <A  HREF="http://BioCyc.org/downloads.shtml">http://BioCyc.org/downloads.shtml</A>
  To obtain the Software-Database Bundle, please contact
  biocyc-info@ai.sri.com to obtain a software license agreement.
    o Universities, non-profit research institutes, government
      laboratories: Local installations of BioCyc are free for
      research purposes.
    o Commercial sites: BioCyc is available for a fee to commercial
      institutions.

SBML MODELS GENERATED FROM THE NON-OPEN DATABASES HAVE RESTRICTIONS ON
PUBLICATION AND REDISTRIBUTION. Please see
<A  HREF="http://biocyc.org/all-reg.shtml">http://biocyc.org/all-reg.shtml</A> for details.


Why did I do this?
==================

For those of you in the SBML community, you may already know that the
collection of BioCyc databases provide "electronic reference sources on
the pathways and genomes of different organisms."  However, you  may not
know that most of the Pathway/Genome databases were computationally
derived using a software program called Pathologic. "The input required by
PathoLogic is an annotated genome for the organism, such as in the form of
a Genbank entry. The output produced by PathoLogic is a new pathway/genome
database for the organism."

For those of you in the Pathway Tools community, you may already think of
the Systems Biology Markup Language as a "a computer-readable format for
representing models of biochemical reaction networks." However, you may
not know that SBML currently supports steady-state models of metabolism
such as Flux Balance Analysis (FBA) and Minimization of Metabolic
Adjustment (MOMA), where the only information required is a stoichimetric
matrix of reactions and metabolites, the relationship of genes to the
reactions their products catalyze, and thermodynamic irreversibility
information to place constraints on the fluxes.

  The reason I am posting to these two communities is to encourage the
pathway-tools users to work towards the goal of creating free and open
pathway/genome databases on the one hand, and to encourage the SBML
community to consider the value of integrating information from the
pathway/genome databases into their own models on the other. By
combining the efforts of these two communities, it may one day be
possible to generate complete and consistent models of metabolism
directly from an annotated genome. Biocyc2SBML is an important initial
step towards that direction.


Sincerely,


Jeremy Zucker
Bioinformatics Specialist
Research Computing Department
Dana-Farber Cancer Institute.




</PRE>

<!--X-Body-of-Message-End-->
<!--X-MsgBody-End-->
<!--X-Follow-Ups-->
<HR>
<!--X-Follow-Ups-End-->
<!--X-References-->
<!--X-References-End-->
<!--X-BotPNI-->
<UL>
<LI>Prev by Date:
<STRONG><A HREF="msg00707.html">RE: [sbml-discuss] RE: [Biospice-sepdtf] clarification    onthegenerated SBML model</A></STRONG>
</LI>
<LI>Prev by thread:
<STRONG><A HREF="msg00706.html">Some old messages coming through</A></STRONG>
</LI>
<LI>Index(es):
<UL>
<LI><A HREF="maillist.html#00708"><STRONG>Date</STRONG></A></LI>
<LI><A HREF="threads.html#00708"><STRONG>Thread</STRONG></A></LI>
</UL>
</LI>
</UL>

<!--X-BotPNI-End-->
<!--X-User-Footer-->
<!--X-User-Footer-End-->
</BODY>
</HTML>
